// shippo/sdk/sdk.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Shippo = void 0;
const hooks_js_1 = require("../hooks/hooks.js");
const config_js_1 = require("../lib/config.js");
const http_js_1 = require("../lib/http.js");
const sdks_js_1 = require("../lib/sdks.js");
const addresses_js_1 = require("./addresses.js");
const batches_js_1 = require("./batches.js");
const carrieraccounts_js_1 = require("./carrieraccounts.js");
const carrierparceltemplates_js_1 = require("./carrierparceltemplates.js");
const customsdeclarations_js_1 = require("./customsdeclarations.js");
const customsitems_js_1 = require("./customsitems.js");
const manifests_js_1 = require("./manifests.js");
const orders_js_1 = require("./orders.js");
const parcels_js_1 = require("./parcels.js");
const pickups_js_1 = require("./pickups.js");
const rates_js_1 = require("./rates.js");
const ratesatcheckout_js_1 = require("./ratesatcheckout.js");
const refunds_js_1 = require("./refunds.js");
const servicegroups_js_1 = require("./servicegroups.js");
const shipments_js_1 = require("./shipments.js");
const shippoaccounts_js_1 = require("./shippoaccounts.js");
const trackingstatus_js_1 = require("./trackingstatus.js");
const transactions_js_1 = require("./transactions.js");
const userparceltemplates_js_1 = require("./userparceltemplates.js");
const webhooks_js_1 = require("./webhooks.js");
class Shippo extends sdks_js_1.ClientSDK {
    constructor(options = {}) {
        const opt = options;
        let hooks;
        if (typeof opt === "object" &&
            opt != null &&
            "hooks" in opt &&
            opt.hooks instanceof hooks_js_1.SDKHooks) {
            hooks = opt.hooks;
        }
        else {
            hooks = new hooks_js_1.SDKHooks();
        }
        super({
            client: options.httpClient || new http_js_1.HTTPClient(),
            baseURL: (0, config_js_1.serverURLFromOptions)(options),
            hooks,
        });
        this.options$ = { ...options, hooks };
        void this.options$;
    }
    get addresses() {
        var _a;
        return ((_a = this._addresses) !== null && _a !== void 0 ? _a : (this._addresses = new addresses_js_1.Addresses(this.options$)));
    }
    get batches() {
        var _a;
        return ((_a = this._batches) !== null && _a !== void 0 ? _a : (this._batches = new batches_js_1.Batches(this.options$)));
    }
    get carrierAccounts() {
        var _a;
        return ((_a = this._carrierAccounts) !== null && _a !== void 0 ? _a : (this._carrierAccounts = new carrieraccounts_js_1.CarrierAccounts(this.options$)));
    }
    get customsDeclarations() {
        var _a;
        return ((_a = this._customsDeclarations) !== null && _a !== void 0 ? _a : (this._customsDeclarations = new customsdeclarations_js_1.CustomsDeclarations(this.options$)));
    }
    get customsItems() {
        var _a;
        return ((_a = this._customsItems) !== null && _a !== void 0 ? _a : (this._customsItems = new customsitems_js_1.CustomsItems(this.options$)));
    }
    get ratesAtCheckout() {
        var _a;
        return ((_a = this._ratesAtCheckout) !== null && _a !== void 0 ? _a : (this._ratesAtCheckout = new ratesatcheckout_js_1.RatesAtCheckout(this.options$)));
    }
    get manifests() {
        var _a;
        return ((_a = this._manifests) !== null && _a !== void 0 ? _a : (this._manifests = new manifests_js_1.Manifests(this.options$)));
    }
    get orders() {
        var _a;
        return ((_a = this._orders) !== null && _a !== void 0 ? _a : (this._orders = new orders_js_1.Orders(this.options$)));
    }
    get carrierParcelTemplates() {
        var _a;
        return ((_a = this._carrierParcelTemplates) !== null && _a !== void 0 ? _a : (this._carrierParcelTemplates = new carrierparceltemplates_js_1.CarrierParcelTemplates(this.options$)));
    }
    get parcels() {
        var _a;
        return ((_a = this._parcels) !== null && _a !== void 0 ? _a : (this._parcels = new parcels_js_1.Parcels(this.options$)));
    }
    get pickups() {
        var _a;
        return ((_a = this._pickups) !== null && _a !== void 0 ? _a : (this._pickups = new pickups_js_1.Pickups(this.options$)));
    }
    get rates() {
        var _a;
        return ((_a = this._rates) !== null && _a !== void 0 ? _a : (this._rates = new rates_js_1.Rates(this.options$)));
    }
    get refunds() {
        var _a;
        return ((_a = this._refunds) !== null && _a !== void 0 ? _a : (this._refunds = new refunds_js_1.Refunds(this.options$)));
    }
    get serviceGroups() {
        var _a;
        return ((_a = this._serviceGroups) !== null && _a !== void 0 ? _a : (this._serviceGroups = new servicegroups_js_1.ServiceGroups(this.options$)));
    }
    get shipments() {
        var _a;
        return ((_a = this._shipments) !== null && _a !== void 0 ? _a : (this._shipments = new shipments_js_1.Shipments(this.options$)));
    }
    get trackingStatus() {
        var _a;
        return ((_a = this._trackingStatus) !== null && _a !== void 0 ? _a : (this._trackingStatus = new trackingstatus_js_1.TrackingStatus(this.options$)));
    }
    get transactions() {
        var _a;
        return ((_a = this._transactions) !== null && _a !== void 0 ? _a : (this._transactions = new transactions_js_1.Transactions(this.options$)));
    }
    get userParcelTemplates() {
        var _a;
        return ((_a = this._userParcelTemplates) !== null && _a !== void 0 ? _a : (this._userParcelTemplates = new userparceltemplates_js_1.UserParcelTemplates(this.options$)));
    }
    get shippoAccounts() {
        var _a;
        return ((_a = this._shippoAccounts) !== null && _a !== void 0 ? _a : (this._shippoAccounts = new shippoaccounts_js_1.ShippoAccounts(this.options$)));
    }
    get webhooks() {
        var _a;
        return ((_a = this._webhooks) !== null && _a !== void 0 ? _a : (this._webhooks = new webhooks_js_1.Webhooks(this.options$)));
    }
}
exports.Shippo = Shippo;
