// shippo/sdk/webhooks.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Webhooks = void 0;
const hooks_js_1 = require("../hooks/hooks.js");
const config_js_1 = require("../lib/config.js");
const encodings_js_1 = require("../lib/encodings.js");
const http_js_1 = require("../lib/http.js");
const schemas$ = __importStar(require("../lib/schemas.js"));
const sdks_js_1 = require("../lib/sdks.js");
const components = __importStar(require("../models/components/index.js"));
const operations = __importStar(require("../models/operations/index.js"));
const z = __importStar(require("zod"));
class Webhooks extends sdks_js_1.ClientSDK {
    constructor(options = {}) {
        const opt = options;
        let hooks;
        if (typeof opt === "object" &&
            opt != null &&
            "hooks" in opt &&
            opt.hooks instanceof hooks_js_1.SDKHooks) {
            hooks = opt.hooks;
        }
        else {
            hooks = new hooks_js_1.SDKHooks();
        }
        super({
            client: options.httpClient || new http_js_1.HTTPClient(),
            baseURL: (0, config_js_1.serverURLFromOptions)(options),
            hooks,
        });
        this.options$ = { ...options, hooks };
        void this.options$;
    }
    /**
     * Create a new webhook
     *
     * @remarks
     * Creates a new webhook to send notifications to a URL when a specific event occurs.
     */
    async createWebhook(request, options) {
        const input$ = request;
        const payload$ = schemas$.parse(input$, (value$) => components.WebhookUpdateRequest$.outboundSchema.parse(value$), "Input validation failed");
        const body$ = (0, encodings_js_1.encodeJSON)("body", payload$, { explode: true });
        const path$ = this.templateURLComponent("/webhooks")();
        const query$ = "";
        const headers$ = new Headers({
            "Content-Type": "application/json",
            Accept: "application/json",
        });
        let security$;
        if (typeof this.options$.apiKeyHeader === "function") {
            security$ = { apiKeyHeader: await this.options$.apiKeyHeader() };
        }
        else if (this.options$.apiKeyHeader) {
            security$ = { apiKeyHeader: this.options$.apiKeyHeader };
        }
        else {
            security$ = {};
        }
        const context = {
            operationID: "createWebhook",
            oAuth2Scopes: [],
            securitySource: this.options$.apiKeyHeader,
        };
        const securitySettings$ = this.resolveGlobalSecurity(security$);
        const request$ = this.createRequest$(context, {
            security: securitySettings$,
            method: "POST",
            path: path$,
            headers: headers$,
            query: query$,
            body: body$,
        }, options);
        const response = await this.do$(request$, { context, errorCodes: ["4XX", "5XX"] });
        const [result$] = await this.matcher()
            .json(201, components.Webhook$)
            .fail(["4XX", "5XX"])
            .match(response);
        return result$;
    }
    /**
     * List all webhooks
     *
     * @remarks
     * Returns a list of all webhooks you have created.
     */
    async listWebhooks(options) {
        const path$ = this.templateURLComponent("/webhooks")();
        const query$ = "";
        const headers$ = new Headers({
            Accept: "application/json",
        });
        let security$;
        if (typeof this.options$.apiKeyHeader === "function") {
            security$ = { apiKeyHeader: await this.options$.apiKeyHeader() };
        }
        else if (this.options$.apiKeyHeader) {
            security$ = { apiKeyHeader: this.options$.apiKeyHeader };
        }
        else {
            security$ = {};
        }
        const context = {
            operationID: "listWebhooks",
            oAuth2Scopes: [],
            securitySource: this.options$.apiKeyHeader,
        };
        const securitySettings$ = this.resolveGlobalSecurity(security$);
        const request$ = this.createRequest$(context, {
            security: securitySettings$,
            method: "GET",
            path: path$,
            headers: headers$,
            query: query$,
        }, options);
        const response = await this.do$(request$, { context, errorCodes: ["4XX", "5XX"] });
        const [result$] = await this.matcher()
            .json(200, components.WebhookPaginatedList$)
            .fail(["4XX", "5XX"])
            .match(response);
        return result$;
    }
    /**
     * Retrieve a specific webhook
     *
     * @remarks
     * Returns the details of a specific webhook using the webhook object ID.
     */
    async getWebhook(webhookId, options) {
        const input$ = {
            webhookId: webhookId,
        };
        const payload$ = schemas$.parse(input$, (value$) => operations.GetWebhookRequest$.outboundSchema.parse(value$), "Input validation failed");
        const body$ = null;
        const pathParams$ = {
            webhookId: (0, encodings_js_1.encodeSimple)("webhookId", payload$.webhookId, {
                explode: false,
                charEncoding: "percent",
            }),
        };
        const path$ = this.templateURLComponent("/webhooks/{webhookId}")(pathParams$);
        const query$ = "";
        const headers$ = new Headers({
            Accept: "application/json",
        });
        let security$;
        if (typeof this.options$.apiKeyHeader === "function") {
            security$ = { apiKeyHeader: await this.options$.apiKeyHeader() };
        }
        else if (this.options$.apiKeyHeader) {
            security$ = { apiKeyHeader: this.options$.apiKeyHeader };
        }
        else {
            security$ = {};
        }
        const context = {
            operationID: "getWebhook",
            oAuth2Scopes: [],
            securitySource: this.options$.apiKeyHeader,
        };
        const securitySettings$ = this.resolveGlobalSecurity(security$);
        const request$ = this.createRequest$(context, {
            security: securitySettings$,
            method: "GET",
            path: path$,
            headers: headers$,
            query: query$,
            body: body$,
        }, options);
        const response = await this.do$(request$, { context, errorCodes: ["4XX", "5XX"] });
        const [result$] = await this.matcher()
            .json(200, components.Webhook$)
            .fail(["4XX", "5XX"])
            .match(response);
        return result$;
    }
    /**
     * Update an existing webhook
     *
     * @remarks
     * Updates an existing webhook using the webhook object ID.
     */
    async updateWebhook(webhookId, webhookUpdateRequest, options) {
        const input$ = {
            webhookId: webhookId,
            webhookUpdateRequest: webhookUpdateRequest,
        };
        const payload$ = schemas$.parse(input$, (value$) => operations.UpdateWebhookRequest$.outboundSchema.parse(value$), "Input validation failed");
        const body$ = (0, encodings_js_1.encodeJSON)("body", payload$.WebhookUpdateRequest, { explode: true });
        const pathParams$ = {
            webhookId: (0, encodings_js_1.encodeSimple)("webhookId", payload$.webhookId, {
                explode: false,
                charEncoding: "percent",
            }),
        };
        const path$ = this.templateURLComponent("/webhooks/{webhookId}")(pathParams$);
        const query$ = "";
        const headers$ = new Headers({
            "Content-Type": "application/json",
            Accept: "application/json",
        });
        let security$;
        if (typeof this.options$.apiKeyHeader === "function") {
            security$ = { apiKeyHeader: await this.options$.apiKeyHeader() };
        }
        else if (this.options$.apiKeyHeader) {
            security$ = { apiKeyHeader: this.options$.apiKeyHeader };
        }
        else {
            security$ = {};
        }
        const context = {
            operationID: "updateWebhook",
            oAuth2Scopes: [],
            securitySource: this.options$.apiKeyHeader,
        };
        const securitySettings$ = this.resolveGlobalSecurity(security$);
        const request$ = this.createRequest$(context, {
            security: securitySettings$,
            method: "PUT",
            path: path$,
            headers: headers$,
            query: query$,
            body: body$,
        }, options);
        const response = await this.do$(request$, { context, errorCodes: ["4XX", "5XX"] });
        const [result$] = await this.matcher()
            .json(200, components.Webhook$)
            .fail(["4XX", "5XX"])
            .match(response);
        return result$;
    }
    /**
     * Delete a specific webhook
     *
     * @remarks
     * Deletes a specific webhook using the webhook object ID.
     */
    async deleteWebhook(webhookId, options) {
        const input$ = {
            webhookId: webhookId,
        };
        const payload$ = schemas$.parse(input$, (value$) => operations.DeleteWebhookRequest$.outboundSchema.parse(value$), "Input validation failed");
        const body$ = null;
        const pathParams$ = {
            webhookId: (0, encodings_js_1.encodeSimple)("webhookId", payload$.webhookId, {
                explode: false,
                charEncoding: "percent",
            }),
        };
        const path$ = this.templateURLComponent("/webhooks/{webhookId}")(pathParams$);
        const query$ = "";
        const headers$ = new Headers({
            Accept: "*/*",
        });
        let security$;
        if (typeof this.options$.apiKeyHeader === "function") {
            security$ = { apiKeyHeader: await this.options$.apiKeyHeader() };
        }
        else if (this.options$.apiKeyHeader) {
            security$ = { apiKeyHeader: this.options$.apiKeyHeader };
        }
        else {
            security$ = {};
        }
        const context = {
            operationID: "deleteWebhook",
            oAuth2Scopes: [],
            securitySource: this.options$.apiKeyHeader,
        };
        const securitySettings$ = this.resolveGlobalSecurity(security$);
        const request$ = this.createRequest$(context, {
            security: securitySettings$,
            method: "DELETE",
            path: path$,
            headers: headers$,
            query: query$,
            body: body$,
        }, options);
        const response = await this.do$(request$, { context, errorCodes: ["4XX", "5XX"] });
        const [result$] = await this.matcher()
            .void(204, z.void())
            .fail(["4XX", "5XX"])
            .match(response);
        return result$;
    }
}
exports.Webhooks = Webhooks;
