// shippo/models/operations/getcustomsitem.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetCustomsItemRequest$ = exports.GetCustomsItemGlobals$ = void 0;
const primitives_js_1 = require("../../lib/primitives.js");
const z = __importStar(require("zod"));
/** @internal */
var GetCustomsItemGlobals$;
(function (GetCustomsItemGlobals$) {
    GetCustomsItemGlobals$.inboundSchema = z
        .object({
        "SHIPPO-API-VERSION": z.string().optional(),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            "SHIPPO-API-VERSION": "shippoApiVersion",
        });
    });
    GetCustomsItemGlobals$.outboundSchema = z
        .object({
        shippoApiVersion: z.string().optional(),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            shippoApiVersion: "SHIPPO-API-VERSION",
        });
    });
})(GetCustomsItemGlobals$ || (exports.GetCustomsItemGlobals$ = GetCustomsItemGlobals$ = {}));
/** @internal */
var GetCustomsItemRequest$;
(function (GetCustomsItemRequest$) {
    GetCustomsItemRequest$.inboundSchema = z
        .object({
        CustomsItemId: z.string(),
        page: z.number().int().default(1),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            CustomsItemId: "customsItemId",
        });
    });
    GetCustomsItemRequest$.outboundSchema = z
        .object({
        customsItemId: z.string(),
        page: z.number().int().default(1),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            customsItemId: "CustomsItemId",
        });
    });
})(GetCustomsItemRequest$ || (exports.GetCustomsItemRequest$ = GetCustomsItemRequest$ = {}));
