// shippo/models/operations/getcustomsdeclaration.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetCustomsDeclarationRequest$ = exports.GetCustomsDeclarationGlobals$ = void 0;
const primitives_js_1 = require("../../lib/primitives.js");
const z = __importStar(require("zod"));
/** @internal */
var GetCustomsDeclarationGlobals$;
(function (GetCustomsDeclarationGlobals$) {
    GetCustomsDeclarationGlobals$.inboundSchema = z
        .object({
        "SHIPPO-API-VERSION": z.string().optional(),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            "SHIPPO-API-VERSION": "shippoApiVersion",
        });
    });
    GetCustomsDeclarationGlobals$.outboundSchema = z
        .object({
        shippoApiVersion: z.string().optional(),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            shippoApiVersion: "SHIPPO-API-VERSION",
        });
    });
})(GetCustomsDeclarationGlobals$ || (exports.GetCustomsDeclarationGlobals$ = GetCustomsDeclarationGlobals$ = {}));
/** @internal */
var GetCustomsDeclarationRequest$;
(function (GetCustomsDeclarationRequest$) {
    GetCustomsDeclarationRequest$.inboundSchema = z
        .object({
        CustomsDeclarationId: z.string(),
        page: z.number().int().default(1),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            CustomsDeclarationId: "customsDeclarationId",
        });
    });
    GetCustomsDeclarationRequest$.outboundSchema = z
        .object({
        customsDeclarationId: z.string(),
        page: z.number().int().default(1),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            customsDeclarationId: "CustomsDeclarationId",
        });
    });
})(GetCustomsDeclarationRequest$ || (exports.GetCustomsDeclarationRequest$ = GetCustomsDeclarationRequest$ = {}));
