// shippo/models/operations/initiateoauth2signin.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.InitiateOauth2SigninResponse$ = exports.InitiateOauth2SigninRequest$ = exports.InitiateOauth2SigninGlobals$ = void 0;
const primitives_js_1 = require("../../lib/primitives.js");
const z = __importStar(require("zod"));
/** @internal */
var InitiateOauth2SigninGlobals$;
(function (InitiateOauth2SigninGlobals$) {
    InitiateOauth2SigninGlobals$.inboundSchema = z
        .object({
        "SHIPPO-API-VERSION": z.string().optional(),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            "SHIPPO-API-VERSION": "shippoApiVersion",
        });
    });
    InitiateOauth2SigninGlobals$.outboundSchema = z
        .object({
        shippoApiVersion: z.string().optional(),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            shippoApiVersion: "SHIPPO-API-VERSION",
        });
    });
})(InitiateOauth2SigninGlobals$ || (exports.InitiateOauth2SigninGlobals$ = InitiateOauth2SigninGlobals$ = {}));
/** @internal */
var InitiateOauth2SigninRequest$;
(function (InitiateOauth2SigninRequest$) {
    InitiateOauth2SigninRequest$.inboundSchema = z
        .object({
        CarrierAccountObjectId: z.string(),
        redirect_uri: z.string(),
        state: z.string().optional(),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            CarrierAccountObjectId: "carrierAccountObjectId",
            redirect_uri: "redirectUri",
        });
    });
    InitiateOauth2SigninRequest$.outboundSchema = z
        .object({
        carrierAccountObjectId: z.string(),
        redirectUri: z.string(),
        state: z.string().optional(),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            carrierAccountObjectId: "CarrierAccountObjectId",
            redirectUri: "redirect_uri",
        });
    });
})(InitiateOauth2SigninRequest$ || (exports.InitiateOauth2SigninRequest$ = InitiateOauth2SigninRequest$ = {}));
/** @internal */
var InitiateOauth2SigninResponse$;
(function (InitiateOauth2SigninResponse$) {
    InitiateOauth2SigninResponse$.inboundSchema = z
        .object({
        Headers: z.record(z.array(z.string())),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            Headers: "headers",
        });
    });
    InitiateOauth2SigninResponse$.outboundSchema = z
        .object({
        headers: z.record(z.array(z.string())),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            headers: "Headers",
        });
    });
})(InitiateOauth2SigninResponse$ || (exports.InitiateOauth2SigninResponse$ = InitiateOauth2SigninResponse$ = {}));
