// shippo/models/operations/listshipments.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListShipmentsRequest$ = exports.ListShipmentsGlobals$ = void 0;
const primitives_js_1 = require("../../lib/primitives.js");
const z = __importStar(require("zod"));
/** @internal */
var ListShipmentsGlobals$;
(function (ListShipmentsGlobals$) {
    ListShipmentsGlobals$.inboundSchema = z
        .object({
        "SHIPPO-API-VERSION": z.string().optional(),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            "SHIPPO-API-VERSION": "shippoApiVersion",
        });
    });
    ListShipmentsGlobals$.outboundSchema = z
        .object({
        shippoApiVersion: z.string().optional(),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            shippoApiVersion: "SHIPPO-API-VERSION",
        });
    });
})(ListShipmentsGlobals$ || (exports.ListShipmentsGlobals$ = ListShipmentsGlobals$ = {}));
/** @internal */
var ListShipmentsRequest$;
(function (ListShipmentsRequest$) {
    ListShipmentsRequest$.inboundSchema = z
        .object({
        page: z.number().int().default(1),
        results: z.number().int().default(25),
        object_created_gt: z.string().optional(),
        object_created_gte: z.string().optional(),
        object_created_lt: z.string().optional(),
        object_created_lte: z.string().optional(),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            object_created_gt: "objectCreatedGt",
            object_created_gte: "objectCreatedGte",
            object_created_lt: "objectCreatedLt",
            object_created_lte: "objectCreatedLte",
        });
    });
    ListShipmentsRequest$.outboundSchema = z
        .object({
        page: z.number().int().default(1),
        results: z.number().int().default(25),
        objectCreatedGt: z.string().optional(),
        objectCreatedGte: z.string().optional(),
        objectCreatedLt: z.string().optional(),
        objectCreatedLte: z.string().optional(),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            objectCreatedGt: "object_created_gt",
            objectCreatedGte: "object_created_gte",
            objectCreatedLt: "object_created_lt",
            objectCreatedLte: "object_created_lte",
        });
    });
})(ListShipmentsRequest$ || (exports.ListShipmentsRequest$ = ListShipmentsRequest$ = {}));
