// shippo/models/components/batchcreaterequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchCreateRequest$ = void 0;
const primitives_js_1 = require("../../lib/primitives.js");
const batchshipmentcreaterequest_js_1 = require("./batchshipmentcreaterequest.js");
const labelfiletypeenum_js_1 = require("./labelfiletypeenum.js");
const z = __importStar(require("zod"));
/** @internal */
var BatchCreateRequest$;
(function (BatchCreateRequest$) {
    BatchCreateRequest$.inboundSchema = z
        .object({
        default_carrier_account: z.string(),
        default_servicelevel_token: z.string(),
        label_filetype: labelfiletypeenum_js_1.LabelFileTypeEnum$.inboundSchema.optional(),
        metadata: z.string().optional(),
        batch_shipments: z.array(batchshipmentcreaterequest_js_1.BatchShipmentCreateRequest$.inboundSchema),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            default_carrier_account: "defaultCarrierAccount",
            default_servicelevel_token: "defaultServicelevelToken",
            label_filetype: "labelFiletype",
            batch_shipments: "batchShipments",
        });
    });
    BatchCreateRequest$.outboundSchema = z
        .object({
        defaultCarrierAccount: z.string(),
        defaultServicelevelToken: z.string(),
        labelFiletype: labelfiletypeenum_js_1.LabelFileTypeEnum$.outboundSchema.optional(),
        metadata: z.string().optional(),
        batchShipments: z.array(batchshipmentcreaterequest_js_1.BatchShipmentCreateRequest$.outboundSchema),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            defaultCarrierAccount: "default_carrier_account",
            defaultServicelevelToken: "default_servicelevel_token",
            labelFiletype: "label_filetype",
            batchShipments: "batch_shipments",
        });
    });
})(BatchCreateRequest$ || (exports.BatchCreateRequest$ = BatchCreateRequest$ = {}));
