// shippo/models/components/batchshipmentcreaterequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchShipmentCreateRequest$ = void 0;
const primitives_js_1 = require("../../lib/primitives.js");
const shipmentcreaterequest_js_1 = require("./shipmentcreaterequest.js");
const z = __importStar(require("zod"));
/** @internal */
var BatchShipmentCreateRequest$;
(function (BatchShipmentCreateRequest$) {
    BatchShipmentCreateRequest$.inboundSchema = z
        .object({
        carrier_account: z.string().optional(),
        metadata: z.string().optional(),
        servicelevel_token: z.string().optional(),
        shipment: shipmentcreaterequest_js_1.ShipmentCreateRequest$.inboundSchema,
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            carrier_account: "carrierAccount",
            servicelevel_token: "servicelevelToken",
        });
    });
    BatchShipmentCreateRequest$.outboundSchema = z
        .object({
        carrierAccount: z.string().optional(),
        metadata: z.string().optional(),
        servicelevelToken: z.string().optional(),
        shipment: shipmentcreaterequest_js_1.ShipmentCreateRequest$.outboundSchema,
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            carrierAccount: "carrier_account",
            servicelevelToken: "servicelevel_token",
        });
    });
})(BatchShipmentCreateRequest$ || (exports.BatchShipmentCreateRequest$ = BatchShipmentCreateRequest$ = {}));
