// shippo/models/components/billing.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Billing$ = exports.Type$ = exports.Type = void 0;
const primitives_js_1 = require("../../lib/primitives.js");
const z = __importStar(require("zod"));
/**
 * Party to be billed. (Leave blank for DHL Germany.)
 */
exports.Type = {
    Sender: "SENDER",
    Recipient: "RECIPIENT",
    ThirdParty: "THIRD_PARTY",
    ThirdPartyConsignee: "THIRD_PARTY_CONSIGNEE",
    Collect: "COLLECT",
};
/** @internal */
var Type$;
(function (Type$) {
    Type$.inboundSchema = z.nativeEnum(exports.Type);
    Type$.outboundSchema = Type$.inboundSchema;
})(Type$ || (exports.Type$ = Type$ = {}));
/** @internal */
var Billing$;
(function (Billing$) {
    Billing$.inboundSchema = z
        .object({
        account: z.string().optional(),
        country: z.string().optional(),
        participation_code: z.string().optional(),
        type: Type$.inboundSchema.optional(),
        zip: z.string().optional(),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            participation_code: "participationCode",
        });
    });
    Billing$.outboundSchema = z
        .object({
        account: z.string().optional(),
        country: z.string().optional(),
        participationCode: z.string().optional(),
        type: Type$.outboundSchema.optional(),
        zip: z.string().optional(),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            participationCode: "participation_code",
        });
    });
})(Billing$ || (exports.Billing$ = Billing$ = {}));
