// shippo/models/components/carrierparceltemplate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CarrierParcelTemplate$ = void 0;
const primitives_js_1 = require("../../lib/primitives.js");
const distanceunitenum_js_1 = require("./distanceunitenum.js");
const z = __importStar(require("zod"));
/** @internal */
var CarrierParcelTemplate$;
(function (CarrierParcelTemplate$) {
    CarrierParcelTemplate$.inboundSchema = z
        .object({
        carrier: z.string().optional(),
        distance_unit: distanceunitenum_js_1.DistanceUnitEnum$.inboundSchema.optional(),
        height: z.string().optional(),
        is_variable_dimensions: z.boolean().optional(),
        length: z.string().optional(),
        name: z.string().optional(),
        token: z.string().optional(),
        width: z.string().optional(),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            distance_unit: "distanceUnit",
            is_variable_dimensions: "isVariableDimensions",
        });
    });
    CarrierParcelTemplate$.outboundSchema = z
        .object({
        carrier: z.string().optional(),
        distanceUnit: distanceunitenum_js_1.DistanceUnitEnum$.outboundSchema.optional(),
        height: z.string().optional(),
        isVariableDimensions: z.boolean().optional(),
        length: z.string().optional(),
        name: z.string().optional(),
        token: z.string().optional(),
        width: z.string().optional(),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            distanceUnit: "distance_unit",
            isVariableDimensions: "is_variable_dimensions",
        });
    });
})(CarrierParcelTemplate$ || (exports.CarrierParcelTemplate$ = CarrierParcelTemplate$ = {}));
