// shippo/models/components/cod.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Cod$ = exports.PaymentMethod$ = exports.PaymentMethod = void 0;
const primitives_js_1 = require("../../lib/primitives.js");
const z = __importStar(require("zod"));
/**
 * Secured funds include money orders, certified cheques and others (see
 *
 * @remarks
 * <a href="https://www.ups.com/content/us/en/shipping/time/service/value_added/cod.html">UPS</a> for details).
 * If no payment_method inputted the value defaults to "ANY".)
 */
exports.PaymentMethod = {
    SecuredFunds: "SECURED_FUNDS",
    Cash: "CASH",
    Any: "ANY",
};
/** @internal */
var PaymentMethod$;
(function (PaymentMethod$) {
    PaymentMethod$.inboundSchema = z.nativeEnum(exports.PaymentMethod);
    PaymentMethod$.outboundSchema = PaymentMethod$.inboundSchema;
})(PaymentMethod$ || (exports.PaymentMethod$ = PaymentMethod$ = {}));
/** @internal */
var Cod$;
(function (Cod$) {
    Cod$.inboundSchema = z
        .object({
        amount: z.string().optional(),
        currency: z.string().optional(),
        payment_method: PaymentMethod$.inboundSchema.optional(),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            payment_method: "paymentMethod",
        });
    });
    Cod$.outboundSchema = z
        .object({
        amount: z.string().optional(),
        currency: z.string().optional(),
        paymentMethod: PaymentMethod$.outboundSchema.optional(),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            paymentMethod: "payment_method",
        });
    });
})(Cod$ || (exports.Cod$ = Cod$ = {}));
