// shippo/models/components/customsdeclaration.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomsDeclaration$ = void 0;
const primitives_js_1 = require("../../lib/primitives.js");
const customsexporteridentification_js_1 = require("./customsexporteridentification.js");
const customsinvoicedcharges_js_1 = require("./customsinvoicedcharges.js");
const objectstateenum_js_1 = require("./objectstateenum.js");
const z = __importStar(require("zod"));
/** @internal */
var CustomsDeclaration$;
(function (CustomsDeclaration$) {
    CustomsDeclaration$.inboundSchema = z
        .object({
        aes_itn: z.string().optional(),
        b13a_filing_option: z.string().optional(),
        b13a_number: z.string().optional(),
        certificate: z.string().optional(),
        certify: z.boolean(),
        certify_signer: z.string(),
        commercial_invoice: z.boolean().optional(),
        contents_explanation: z.string().optional(),
        disclaimer: z.string().optional(),
        exporter_identification: customsexporteridentification_js_1.CustomsExporterIdentification$.inboundSchema.optional(),
        exporter_reference: z.string().optional(),
        importer_reference: z.string().optional(),
        is_vat_collected: z.boolean().optional(),
        invoice: z.string().optional(),
        license: z.string().optional(),
        metadata: z.string().optional(),
        notes: z.string().optional(),
        address_importer: z.string().optional(),
        contents_type: z.string(),
        eel_pfc: z.string().optional(),
        incoterm: z.string().optional(),
        invoiced_charges: customsinvoicedcharges_js_1.CustomsInvoicedCharges$.inboundSchema.optional(),
        items: z.array(z.string()),
        non_delivery_option: z.string(),
        object_created: z
            .string()
            .datetime({ offset: true })
            .transform((v) => new Date(v))
            .optional(),
        object_id: z.string().optional(),
        object_owner: z.string().optional(),
        object_state: objectstateenum_js_1.ObjectStateEnum$.inboundSchema.optional(),
        object_updated: z
            .string()
            .datetime({ offset: true })
            .transform((v) => new Date(v))
            .optional(),
        test: z.boolean().optional(),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            aes_itn: "aesItn",
            b13a_filing_option: "b13aFilingOption",
            b13a_number: "b13aNumber",
            certify_signer: "certifySigner",
            commercial_invoice: "commercialInvoice",
            contents_explanation: "contentsExplanation",
            exporter_identification: "exporterIdentification",
            exporter_reference: "exporterReference",
            importer_reference: "importerReference",
            is_vat_collected: "isVatCollected",
            address_importer: "addressImporter",
            contents_type: "contentsType",
            eel_pfc: "eelPfc",
            invoiced_charges: "invoicedCharges",
            non_delivery_option: "nonDeliveryOption",
            object_created: "objectCreated",
            object_id: "objectId",
            object_owner: "objectOwner",
            object_state: "objectState",
            object_updated: "objectUpdated",
        });
    });
    CustomsDeclaration$.outboundSchema = z
        .object({
        aesItn: z.string().optional(),
        b13aFilingOption: z.string().optional(),
        b13aNumber: z.string().optional(),
        certificate: z.string().optional(),
        certify: z.boolean(),
        certifySigner: z.string(),
        commercialInvoice: z.boolean().optional(),
        contentsExplanation: z.string().optional(),
        disclaimer: z.string().optional(),
        exporterIdentification: customsexporteridentification_js_1.CustomsExporterIdentification$.outboundSchema.optional(),
        exporterReference: z.string().optional(),
        importerReference: z.string().optional(),
        isVatCollected: z.boolean().optional(),
        invoice: z.string().optional(),
        license: z.string().optional(),
        metadata: z.string().optional(),
        notes: z.string().optional(),
        addressImporter: z.string().optional(),
        contentsType: z.string(),
        eelPfc: z.string().optional(),
        incoterm: z.string().optional(),
        invoicedCharges: customsinvoicedcharges_js_1.CustomsInvoicedCharges$.outboundSchema.optional(),
        items: z.array(z.string()),
        nonDeliveryOption: z.string(),
        objectCreated: z
            .date()
            .transform((v) => v.toISOString())
            .optional(),
        objectId: z.string().optional(),
        objectOwner: z.string().optional(),
        objectState: objectstateenum_js_1.ObjectStateEnum$.outboundSchema.optional(),
        objectUpdated: z
            .date()
            .transform((v) => v.toISOString())
            .optional(),
        test: z.boolean().optional(),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            aesItn: "aes_itn",
            b13aFilingOption: "b13a_filing_option",
            b13aNumber: "b13a_number",
            certifySigner: "certify_signer",
            commercialInvoice: "commercial_invoice",
            contentsExplanation: "contents_explanation",
            exporterIdentification: "exporter_identification",
            exporterReference: "exporter_reference",
            importerReference: "importer_reference",
            isVatCollected: "is_vat_collected",
            addressImporter: "address_importer",
            contentsType: "contents_type",
            eelPfc: "eel_pfc",
            invoicedCharges: "invoiced_charges",
            nonDeliveryOption: "non_delivery_option",
            objectCreated: "object_created",
            objectId: "object_id",
            objectOwner: "object_owner",
            objectState: "object_state",
            objectUpdated: "object_updated",
        });
    });
})(CustomsDeclaration$ || (exports.CustomsDeclaration$ = CustomsDeclaration$ = {}));
