// shippo/models/components/insurance.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Insurance$ = exports.Provider$ = exports.Provider = void 0;
const z = __importStar(require("zod"));
/**
 * To have insurance cover provided by a carrier directly instead of Shippo's provider (XCover), set `provider` to `FEDEX`, `UPS`, or `ONTRAC`.
 */
exports.Provider = {
    Fedex: "FEDEX",
    Ups: "UPS",
    Ontrac: "ONTRAC",
};
/** @internal */
var Provider$;
(function (Provider$) {
    Provider$.inboundSchema = z.nativeEnum(exports.Provider);
    Provider$.outboundSchema = Provider$.inboundSchema;
})(Provider$ || (exports.Provider$ = Provider$ = {}));
/** @internal */
var Insurance$;
(function (Insurance$) {
    Insurance$.inboundSchema = z.object({
        amount: z.string().optional(),
        content: z.string().optional(),
        currency: z.string().optional(),
        provider: Provider$.inboundSchema.optional(),
    });
    Insurance$.outboundSchema = z.object({
        amount: z.string().optional(),
        content: z.string().optional(),
        currency: z.string().optional(),
        provider: Provider$.outboundSchema.optional(),
    });
})(Insurance$ || (exports.Insurance$ = Insurance$ = {}));
