// shippo/models/components/labelfiletypeenum.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LabelFileTypeEnum$ = exports.LabelFileTypeEnum = void 0;
const z = __importStar(require("zod"));
/**
 * Print format of the <a href="https://docs.goshippo.com/docs/shipments/shippinglabelsizes/">label</a>. If empty, will use the default format set from
 *
 * @remarks
 * <a href="https://apps.goshippo.com/settings/labels">the Shippo dashboard.</a>
 */
exports.LabelFileTypeEnum = {
    Png: "PNG",
    PNG23x75: "PNG_2.3x7.5",
    Pdf: "PDF",
    PDF23x75: "PDF_2.3x7.5",
    PDF4x6: "PDF_4x6",
    PDF4x8: "PDF_4x8",
    PdfA4: "PDF_A4",
    PdfA5: "PDF_A5",
    PdfA6: "PDF_A6",
    Zplii: "ZPLII",
};
/** @internal */
var LabelFileTypeEnum$;
(function (LabelFileTypeEnum$) {
    LabelFileTypeEnum$.inboundSchema = z.nativeEnum(exports.LabelFileTypeEnum);
    LabelFileTypeEnum$.outboundSchema = LabelFileTypeEnum$.inboundSchema;
})(LabelFileTypeEnum$ || (exports.LabelFileTypeEnum$ = LabelFileTypeEnum$ = {}));
