// shippo/models/components/liveratecreaterequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LiveRateCreateRequest$ = exports.LiveRateCreateRequestParcel$ = exports.LiveRateCreateRequestAddressTo$ = exports.LiveRateCreateRequestAddressFrom$ = void 0;
const primitives_js_1 = require("../../lib/primitives.js");
const addresscompletecreaterequest_js_1 = require("./addresscompletecreaterequest.js");
const lineitem_js_1 = require("./lineitem.js");
const parcel_js_1 = require("./parcel.js");
const z = __importStar(require("zod"));
/** @internal */
var LiveRateCreateRequestAddressFrom$;
(function (LiveRateCreateRequestAddressFrom$) {
    LiveRateCreateRequestAddressFrom$.inboundSchema = z.union([addresscompletecreaterequest_js_1.AddressCompleteCreateRequest$.inboundSchema, z.string()]);
    LiveRateCreateRequestAddressFrom$.outboundSchema = z.union([addresscompletecreaterequest_js_1.AddressCompleteCreateRequest$.outboundSchema, z.string()]);
})(LiveRateCreateRequestAddressFrom$ || (exports.LiveRateCreateRequestAddressFrom$ = LiveRateCreateRequestAddressFrom$ = {}));
/** @internal */
var LiveRateCreateRequestAddressTo$;
(function (LiveRateCreateRequestAddressTo$) {
    LiveRateCreateRequestAddressTo$.inboundSchema = z.union([addresscompletecreaterequest_js_1.AddressCompleteCreateRequest$.inboundSchema, z.string()]);
    LiveRateCreateRequestAddressTo$.outboundSchema = z.union([addresscompletecreaterequest_js_1.AddressCompleteCreateRequest$.outboundSchema, z.string()]);
})(LiveRateCreateRequestAddressTo$ || (exports.LiveRateCreateRequestAddressTo$ = LiveRateCreateRequestAddressTo$ = {}));
/** @internal */
var LiveRateCreateRequestParcel$;
(function (LiveRateCreateRequestParcel$) {
    LiveRateCreateRequestParcel$.inboundSchema = z.union([parcel_js_1.Parcel$.inboundSchema, z.string()]);
    LiveRateCreateRequestParcel$.outboundSchema = z.union([parcel_js_1.Parcel$.outboundSchema, z.string()]);
})(LiveRateCreateRequestParcel$ || (exports.LiveRateCreateRequestParcel$ = LiveRateCreateRequestParcel$ = {}));
/** @internal */
var LiveRateCreateRequest$;
(function (LiveRateCreateRequest$) {
    LiveRateCreateRequest$.inboundSchema = z
        .object({
        address_from: z
            .union([addresscompletecreaterequest_js_1.AddressCompleteCreateRequest$.inboundSchema, z.string()])
            .optional(),
        address_to: z.union([addresscompletecreaterequest_js_1.AddressCompleteCreateRequest$.inboundSchema, z.string()]),
        line_items: z.array(lineitem_js_1.LineItem$.inboundSchema),
        parcel: z.union([parcel_js_1.Parcel$.inboundSchema, z.string()]).optional(),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            address_from: "addressFrom",
            address_to: "addressTo",
            line_items: "lineItems",
        });
    });
    LiveRateCreateRequest$.outboundSchema = z
        .object({
        addressFrom: z
            .union([addresscompletecreaterequest_js_1.AddressCompleteCreateRequest$.outboundSchema, z.string()])
            .optional(),
        addressTo: z.union([addresscompletecreaterequest_js_1.AddressCompleteCreateRequest$.outboundSchema, z.string()]),
        lineItems: z.array(lineitem_js_1.LineItem$.outboundSchema),
        parcel: z.union([parcel_js_1.Parcel$.outboundSchema, z.string()]).optional(),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            addressFrom: "address_from",
            addressTo: "address_to",
            lineItems: "line_items",
        });
    });
})(LiveRateCreateRequest$ || (exports.LiveRateCreateRequest$ = LiveRateCreateRequest$ = {}));
