// shippo/models/components/liverate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LiveRate$ = void 0;
const primitives_js_1 = require("../../lib/primitives.js");
const z = __importStar(require("zod"));
/** @internal */
var LiveRate$;
(function (LiveRate$) {
    LiveRate$.inboundSchema = z
        .object({
        amount: z.string().optional(),
        amount_local: z.string().optional(),
        currency: z.string().optional(),
        currency_local: z.string().optional(),
        estimated_days: z.number().int().optional(),
        title: z.string().optional(),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            amount_local: "amountLocal",
            currency_local: "currencyLocal",
            estimated_days: "estimatedDays",
        });
    });
    LiveRate$.outboundSchema = z
        .object({
        amount: z.string().optional(),
        amountLocal: z.string().optional(),
        currency: z.string().optional(),
        currencyLocal: z.string().optional(),
        estimatedDays: z.number().int().optional(),
        title: z.string().optional(),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            amountLocal: "amount_local",
            currencyLocal: "currency_local",
            estimatedDays: "estimated_days",
        });
    });
})(LiveRate$ || (exports.LiveRate$ = LiveRate$ = {}));
