// shippo/models/components/location.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Location$ = exports.BuildingType$ = exports.BuildingLocationType$ = exports.BuildingType = exports.BuildingLocationType = void 0;
const primitives_js_1 = require("../../lib/primitives.js");
const addresscompletecreaterequest_js_1 = require("./addresscompletecreaterequest.js");
const z = __importStar(require("zod"));
/**
 * Where your parcels will be available for pickup. "Security Deck" and "Shipping Dock" are only
 *
 * @remarks
 * supported for DHL Express.
 */
exports.BuildingLocationType = {
    BackDoor: "Back Door",
    RingBell: "Ring Bell",
    SecurityDeck: "Security Deck",
    ShippingDock: "Shipping Dock",
    FrontDoor: "Front Door",
    KnockOnDoor: "Knock on Door",
    InAtMailbox: "In/At Mailbox",
    MailRoom: "Mail Room",
    Office: "Office",
    Other: "Other",
    Reception: "Reception",
    SideDoor: "Side Door",
};
/**
 * The type of building where the pickup is located.
 */
exports.BuildingType = {
    Apartment: "apartment",
    Building: "building",
    Department: "department",
    Floor: "floor",
    Room: "room",
    Suite: "suite",
};
/** @internal */
var BuildingLocationType$;
(function (BuildingLocationType$) {
    BuildingLocationType$.inboundSchema = z.nativeEnum(exports.BuildingLocationType);
    BuildingLocationType$.outboundSchema = BuildingLocationType$.inboundSchema;
})(BuildingLocationType$ || (exports.BuildingLocationType$ = BuildingLocationType$ = {}));
/** @internal */
var BuildingType$;
(function (BuildingType$) {
    BuildingType$.inboundSchema = z.nativeEnum(exports.BuildingType);
    BuildingType$.outboundSchema = BuildingType$.inboundSchema;
})(BuildingType$ || (exports.BuildingType$ = BuildingType$ = {}));
/** @internal */
var Location$;
(function (Location$) {
    Location$.inboundSchema = z
        .object({
        address: addresscompletecreaterequest_js_1.AddressCompleteCreateRequest$.inboundSchema,
        building_location_type: BuildingLocationType$.inboundSchema,
        building_type: BuildingType$.inboundSchema.optional(),
        instructions: z.string().optional(),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            building_location_type: "buildingLocationType",
            building_type: "buildingType",
        });
    });
    Location$.outboundSchema = z
        .object({
        address: addresscompletecreaterequest_js_1.AddressCompleteCreateRequest$.outboundSchema,
        buildingLocationType: BuildingLocationType$.outboundSchema,
        buildingType: BuildingType$.outboundSchema.optional(),
        instructions: z.string().optional(),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            buildingLocationType: "building_location_type",
            buildingType: "building_type",
        });
    });
})(Location$ || (exports.Location$ = Location$ = {}));
