// shippo/models/components/manifestcreaterequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ManifestCreateRequest$ = exports.ManifestCreateRequestAddressFrom$ = void 0;
const primitives_js_1 = require("../../lib/primitives.js");
const addresscreaterequest_js_1 = require("./addresscreaterequest.js");
const z = __importStar(require("zod"));
/** @internal */
var ManifestCreateRequestAddressFrom$;
(function (ManifestCreateRequestAddressFrom$) {
    ManifestCreateRequestAddressFrom$.inboundSchema = z.union([addresscreaterequest_js_1.AddressCreateRequest$.inboundSchema, z.string()]);
    ManifestCreateRequestAddressFrom$.outboundSchema = z.union([addresscreaterequest_js_1.AddressCreateRequest$.outboundSchema, z.string()]);
})(ManifestCreateRequestAddressFrom$ || (exports.ManifestCreateRequestAddressFrom$ = ManifestCreateRequestAddressFrom$ = {}));
/** @internal */
var ManifestCreateRequest$;
(function (ManifestCreateRequest$) {
    ManifestCreateRequest$.inboundSchema = z
        .object({
        carrier_account: z.string(),
        shipment_date: z.string(),
        transactions: z.array(z.string()).optional(),
        address_from: z.union([addresscreaterequest_js_1.AddressCreateRequest$.inboundSchema, z.string()]),
        async: z.boolean().optional(),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            carrier_account: "carrierAccount",
            shipment_date: "shipmentDate",
            address_from: "addressFrom",
        });
    });
    ManifestCreateRequest$.outboundSchema = z
        .object({
        carrierAccount: z.string(),
        shipmentDate: z.string(),
        transactions: z.array(z.string()).optional(),
        addressFrom: z.union([addresscreaterequest_js_1.AddressCreateRequest$.outboundSchema, z.string()]),
        async: z.boolean().optional(),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            carrierAccount: "carrier_account",
            shipmentDate: "shipment_date",
            addressFrom: "address_from",
        });
    });
})(ManifestCreateRequest$ || (exports.ManifestCreateRequest$ = ManifestCreateRequest$ = {}));
