// shippo/models/components/parcel.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Parcel$ = exports.ObjectState$ = exports.ObjectState = void 0;
const primitives_js_1 = require("../../lib/primitives.js");
const distanceunitenum_js_1 = require("./distanceunitenum.js");
const parcelextra_js_1 = require("./parcelextra.js");
const parceltemplateenumset_js_1 = require("./parceltemplateenumset.js");
const weightunitenum_js_1 = require("./weightunitenum.js");
const z = __importStar(require("zod"));
/**
 * A Parcel will only be valid when all required values have been sent and validated successfully.
 */
exports.ObjectState = {
    Valid: "VALID",
};
/** @internal */
var ObjectState$;
(function (ObjectState$) {
    ObjectState$.inboundSchema = z.nativeEnum(exports.ObjectState);
    ObjectState$.outboundSchema = ObjectState$.inboundSchema;
})(ObjectState$ || (exports.ObjectState$ = ObjectState$ = {}));
/** @internal */
var Parcel$;
(function (Parcel$) {
    Parcel$.inboundSchema = z
        .object({
        extra: parcelextra_js_1.ParcelExtra$.inboundSchema.optional(),
        metadata: z.string().optional(),
        mass_unit: weightunitenum_js_1.WeightUnitEnum$.inboundSchema,
        weight: z.string(),
        distance_unit: distanceunitenum_js_1.DistanceUnitEnum$.inboundSchema,
        height: z.string(),
        length: z.string(),
        width: z.string(),
        object_created: z
            .string()
            .datetime({ offset: true })
            .transform((v) => new Date(v))
            .optional(),
        object_id: z.string().optional(),
        object_owner: z.string().optional(),
        object_state: ObjectState$.inboundSchema.optional(),
        object_updated: z
            .string()
            .datetime({ offset: true })
            .transform((v) => new Date(v))
            .optional(),
        template: parceltemplateenumset_js_1.ParcelTemplateEnumSet$.inboundSchema.optional(),
        test: z.boolean().optional(),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            mass_unit: "massUnit",
            distance_unit: "distanceUnit",
            object_created: "objectCreated",
            object_id: "objectId",
            object_owner: "objectOwner",
            object_state: "objectState",
            object_updated: "objectUpdated",
        });
    });
    Parcel$.outboundSchema = z
        .object({
        extra: parcelextra_js_1.ParcelExtra$.outboundSchema.optional(),
        metadata: z.string().optional(),
        massUnit: weightunitenum_js_1.WeightUnitEnum$.outboundSchema,
        weight: z.string(),
        distanceUnit: distanceunitenum_js_1.DistanceUnitEnum$.outboundSchema,
        height: z.string(),
        length: z.string(),
        width: z.string(),
        objectCreated: z
            .date()
            .transform((v) => v.toISOString())
            .optional(),
        objectId: z.string().optional(),
        objectOwner: z.string().optional(),
        objectState: ObjectState$.outboundSchema.optional(),
        objectUpdated: z
            .date()
            .transform((v) => v.toISOString())
            .optional(),
        template: parceltemplateenumset_js_1.ParcelTemplateEnumSet$.outboundSchema.optional(),
        test: z.boolean().optional(),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            massUnit: "mass_unit",
            distanceUnit: "distance_unit",
            objectCreated: "object_created",
            objectId: "object_id",
            objectOwner: "object_owner",
            objectState: "object_state",
            objectUpdated: "object_updated",
        });
    });
})(Parcel$ || (exports.Parcel$ = Parcel$ = {}));
