// shippo/models/components/parcelcreaterequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ParcelCreateRequest$ = void 0;
const primitives_js_1 = require("../../lib/primitives.js");
const distanceunitenum_js_1 = require("./distanceunitenum.js");
const parcelextra_js_1 = require("./parcelextra.js");
const weightunitenum_js_1 = require("./weightunitenum.js");
const z = __importStar(require("zod"));
/** @internal */
var ParcelCreateRequest$;
(function (ParcelCreateRequest$) {
    ParcelCreateRequest$.inboundSchema = z
        .object({
        extra: parcelextra_js_1.ParcelExtra$.inboundSchema.optional(),
        metadata: z.string().optional(),
        mass_unit: weightunitenum_js_1.WeightUnitEnum$.inboundSchema,
        weight: z.string(),
        distance_unit: distanceunitenum_js_1.DistanceUnitEnum$.inboundSchema,
        height: z.string(),
        length: z.string(),
        width: z.string(),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            mass_unit: "massUnit",
            distance_unit: "distanceUnit",
        });
    });
    ParcelCreateRequest$.outboundSchema = z
        .object({
        extra: parcelextra_js_1.ParcelExtra$.outboundSchema.optional(),
        metadata: z.string().optional(),
        massUnit: weightunitenum_js_1.WeightUnitEnum$.outboundSchema,
        weight: z.string(),
        distanceUnit: distanceunitenum_js_1.DistanceUnitEnum$.outboundSchema,
        height: z.string(),
        length: z.string(),
        width: z.string(),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            massUnit: "mass_unit",
            distanceUnit: "distance_unit",
        });
    });
})(ParcelCreateRequest$ || (exports.ParcelCreateRequest$ = ParcelCreateRequest$ = {}));
