// shippo/models/components/pickupbase.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PickupBase$ = void 0;
const primitives_js_1 = require("../../lib/primitives.js");
const location_js_1 = require("./location.js");
const z = __importStar(require("zod"));
/** @internal */
var PickupBase$;
(function (PickupBase$) {
    PickupBase$.inboundSchema = z
        .object({
        carrier_account: z.string(),
        location: location_js_1.Location$.inboundSchema,
        metadata: z.string().optional(),
        requested_end_time: z
            .string()
            .datetime({ offset: true })
            .transform((v) => new Date(v)),
        requested_start_time: z
            .string()
            .datetime({ offset: true })
            .transform((v) => new Date(v)),
        transactions: z.array(z.string()),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            carrier_account: "carrierAccount",
            requested_end_time: "requestedEndTime",
            requested_start_time: "requestedStartTime",
        });
    });
    PickupBase$.outboundSchema = z
        .object({
        carrierAccount: z.string(),
        location: location_js_1.Location$.outboundSchema,
        metadata: z.string().optional(),
        requestedEndTime: z.date().transform((v) => v.toISOString()),
        requestedStartTime: z.date().transform((v) => v.toISOString()),
        transactions: z.array(z.string()),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            carrierAccount: "carrier_account",
            requestedEndTime: "requested_end_time",
            requestedStartTime: "requested_start_time",
        });
    });
})(PickupBase$ || (exports.PickupBase$ = PickupBase$ = {}));
