// shippo/models/components/servicelevelenumset.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceLevelePostGlobalEnum$ = exports.ServiceLevelRoyalMailEnum$ = exports.ServiceLevelSendleEnum$ = exports.ServiceLevelOnTracEnum$ = exports.ServiceLevelJitsuEnum$ = exports.ServiceLevelLasershipEnum$ = exports.ServiceLevelEvriUKEnum$ = exports.ServiceLevelUDSEnum$ = exports.ServiceLevelVehoEnum$ = exports.ServiceLevelSwyftEnum$ = exports.ServiceLevelUSPSEnum = exports.ServiceLevelFedExEnum = exports.ServiceLevelUPSEnum = exports.ServiceLevelAirterraEnum = exports.ServiceLevelAPCPostalEnum = exports.ServiceLevelAsendiaEnum = exports.ServiceLevelAustraliaPostEnum = exports.ServiceLevelAPGEnum = exports.ServiceLevelBetterTrucksEnum = exports.ServiceLevelCanadaPostEnum = exports.ServiceLevelCDLEnum = exports.ServiceLevelChronopostEnum = exports.ServiceLevelCouriersPleaseEnum = exports.ServiceLevelCorreosEspanaEnum = exports.ServiceLevelColissimoEnum = exports.ServiceLevelPurolatorEnum = exports.ServiceLevelDHLExpressEnum = exports.ServiceLevelDHLeCommerceEnum = exports.ServiceLevelDHLGermanyEnum = exports.ServiceLevelDPDDEEnum = exports.ServiceLevelDPDUKEnum = exports.ServiceLevelDeutschePostEnum = exports.ServiceLevelAramexAustraliaEnum = exports.ServiceLevelGlobegisticsEnum = exports.ServiceLevelGLSUSEnum = exports.ServiceLevelLSOEnum = exports.ServiceLevelMondialRelayEnum = exports.ServiceLevelMaergoEnum = exports.ServiceLevelParcelforceEnum = exports.ServiceLevelPostItalianeEnum = exports.ServiceLevelePostGlobalEnum = exports.ServiceLevelRoyalMailEnum = exports.ServiceLevelSendleEnum = exports.ServiceLevelOnTracEnum = exports.ServiceLevelJitsuEnum = exports.ServiceLevelLasershipEnum = exports.ServiceLevelEvriUKEnum = exports.ServiceLevelUDSEnum = exports.ServiceLevelVehoEnum = exports.ServiceLevelSwyftEnum = void 0;
exports.ServiceLevelEnumSet$ = exports.ServiceLevelUSPSEnum$ = exports.ServiceLevelFedExEnum$ = exports.ServiceLevelUPSEnum$ = exports.ServiceLevelAirterraEnum$ = exports.ServiceLevelAPCPostalEnum$ = exports.ServiceLevelAsendiaEnum$ = exports.ServiceLevelAustraliaPostEnum$ = exports.ServiceLevelAPGEnum$ = exports.ServiceLevelBetterTrucksEnum$ = exports.ServiceLevelCanadaPostEnum$ = exports.ServiceLevelCDLEnum$ = exports.ServiceLevelChronopostEnum$ = exports.ServiceLevelCouriersPleaseEnum$ = exports.ServiceLevelCorreosEspanaEnum$ = exports.ServiceLevelColissimoEnum$ = exports.ServiceLevelPurolatorEnum$ = exports.ServiceLevelDHLExpressEnum$ = exports.ServiceLevelDHLeCommerceEnum$ = exports.ServiceLevelDHLGermanyEnum$ = exports.ServiceLevelDPDDEEnum$ = exports.ServiceLevelDPDUKEnum$ = exports.ServiceLevelDeutschePostEnum$ = exports.ServiceLevelAramexAustraliaEnum$ = exports.ServiceLevelGlobegisticsEnum$ = exports.ServiceLevelGLSUSEnum$ = exports.ServiceLevelLSOEnum$ = exports.ServiceLevelMondialRelayEnum$ = exports.ServiceLevelMaergoEnum$ = exports.ServiceLevelParcelforceEnum$ = exports.ServiceLevelPostItalianeEnum$ = void 0;
const z = __importStar(require("zod"));
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | same_day | Next Day|
 * | next_day | Next Day|
 *
 */
exports.ServiceLevelSwyftEnum = {
    SameDay: "same_day",
    NextDay: "next_day",
};
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | veho_next_day | Veho Next Day |
 *
 */
exports.ServiceLevelVehoEnum = {
    VehoNextDay: "veho_next_day",
};
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | uds_next_day | Next Day|
 *
 */
exports.ServiceLevelUDSEnum = {
    UdsNextDay: "uds_next_day",
};
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | hermes_uk_courier_service | Courier Collection|
 * | hermes_uk_parcelshop_dropoff | ParcelShop Drop-Off|
 * | hermes_uk_parcelshop_dropoff_nextday | ParcelShop Drop-Off Next Day|
 * | hermes_uk_postable | Postable|
 * | hermes_uk_postable_nextday | Postable Next Day|
 *
 */
exports.ServiceLevelEvriUKEnum = {
    HermesUkCourierService: "hermes_uk_courier_service",
    HermesUkParcelshopDropoff: "hermes_uk_parcelshop_dropoff",
    HermesUkParcelshopDropoffNextday: "hermes_uk_parcelshop_dropoff_nextday",
    HermesUkPostable: "hermes_uk_postable",
    HermesUkPostableNextday: "hermes_uk_postable_nextday",
};
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | lasership_routed_delivery | Routed Delivery|
 *
 */
exports.ServiceLevelLasershipEnum = {
    LasershipRoutedDelivery: "lasership_routed_delivery",
};
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | axlehire_same_day | Same Day|
 * | axlehire_next_day | Next Day|
 *
 */
exports.ServiceLevelJitsuEnum = {
    AxlehireSameDay: "axlehire_same_day",
    AxlehireNextDay: "axlehire_next_day",
};
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | ontrac_ground | Ground|
 * | ontrac_sunrise_gold | Sunrise Gold|
 * | ontrac_sunrise | Sunrise|
 *
 */
exports.ServiceLevelOnTracEnum = {
    OntracGround: "ontrac_ground",
    OntracSunriseGold: "ontrac_sunrise_gold",
    OntracSunrise: "ontrac_sunrise",
};
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | sendle_parcel | Sendle Parcel|
 * | sendle_standard_dropoff | Sendle Standard Dropoff|
 *
 */
exports.ServiceLevelSendleEnum = {
    SendleParcel: "sendle_parcel",
    SendleStandardDropoff: "sendle_standard_dropoff",
};
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | royal_mail_tracked_letter_boxable_24_no_signature | Royal Mail Tracked Letter-Boxable 24 No Signature|
 * | royal_mail_tracked_letter_boxable_48_no_signature | Royal Mail Tracked Letter-Boxable 48 No Signature|
 * | royal_mail_tracked_24_returns | Royal Mail Tracked Returns 24|
 * | royal_mail_tracked_48_returns | Royal Mail Tracked Returns 48|
 * | royal_mail_special_delivery_guaranteed_1pm | Special Delivery Guaranteed by 1pm
 * | royal_mail_special_delivery_guaranteed_9am | Special Delivery Guaranteed by 9am
 * | royal_mail_24 | RoyalMail 24
 * | royal_mail_48 | RoyalMail 48
 * | royal_mail_signed_for_first_class | Signed for First Class
 * | royal_mail_signed_for_second_class | Signed for Second Class
 * | royal_mail_standard_first_class | Standard First Class
 * | royal_mail_standard_second_class | Standard Second Class
 * | royal_mail_tracked_24_no_signature | Royal Mail Tracked 24 No Signature
 * | royal_mail_tracked_48_no_signature | Royal Mail Ttacked 48 No Signature
 * | royal_mail_intl_bus_mail_lrg_ltr_zone_sort_pri | International Business Mail Large Letter Zone Sort Priority
 * | royal_mail_intl_bus_parcels_tracked_zone_sort | International Business Parcels Tracked Zone Sort
 * | royal_mail_intl_bus_parcels_tracked_country_priced | International Business Parcels Tracked Country Priced
 * | royal_mail_intl_bus_parcels_tracked_signed_zone_srt | International Business Parcels & Signed Zone Sort
 * | royal_mail_24_flat_rate | Royal Mail 24 Parcel Flat Rate
 * | royal_mail_48_flat_rate | Royal Mail 48 Parcel Flat Rate
 *
 */
exports.ServiceLevelRoyalMailEnum = {
    RoyalMailTrackedLetterBoxable24NoSignature: "royal_mail_tracked_letter_boxable_24_no_signature",
    RoyalMailTrackedLetterBoxable48NoSignature: "royal_mail_tracked_letter_boxable_48_no_signature",
    RoyalMailTracked24Returns: "royal_mail_tracked_24_returns",
    RoyalMailTracked48Returns: "royal_mail_tracked_48_returns",
    RoyalMailSpecialDeliveryGuaranteed1pm: "royal_mail_special_delivery_guaranteed_1pm",
    RoyalMailSpecialDeliveryGuaranteed9am: "royal_mail_special_delivery_guaranteed_9am",
    RoyalMail24: "royal_mail_24",
    RoyalMail48: "royal_mail_48",
    RoyalMailSignedForFirstClass: "royal_mail_signed_for_first_class",
    RoyalMailSignedForSecondClass: "royal_mail_signed_for_second_class",
    RoyalMailStandardFirstClass: "royal_mail_standard_first_class",
    RoyalMailStandardSecondClass: "royal_mail_standard_second_class",
    RoyalMailTracked24NoSignature: "royal_mail_tracked_24_no_signature",
    RoyalMailTracked48NoSignature: "royal_mail_tracked_48_no_signature",
    RoyalMailIntlBusMailLrgLtrZoneSortPri: "royal_mail_intl_bus_mail_lrg_ltr_zone_sort_pri",
    RoyalMailIntlBusParcelsTrackedZoneSort: "royal_mail_intl_bus_parcels_tracked_zone_sort",
    RoyalMailIntlBusParcelsTrackedCountryPriced: "royal_mail_intl_bus_parcels_tracked_country_priced",
    RoyalMailIntlBusParcelsTrackedSignedZoneSrt: "royal_mail_intl_bus_parcels_tracked_signed_zone_srt",
    RoyalMail24FlatRate: "royal_mail_24_flat_rate",
    RoyalMail48FlatRate: "royal_mail_48_flat_rate",
};
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | rr_donnelley_domestic_economy_parcel | Domestic Economy Parcel|
 * | rr_donnelley_domestic_priority_parcel | Domestic Priority Parcel |
 * | rr_donnelley_domestic_parcel_bpm | Domestic Parcel BPM|
 * | rr_donnelley_priority_domestic_priority_parcel_bpm | Domestic Priority Parcel BPM|
 * | rr_donnelley_priority_parcel_delcon | International Priority Parcel DelCon|
 * | rr_donnelley_priority_parcel_nondelcon | International Priority Parcel NonDelcon|
 * | rr_donnelley_economy_parcel | Economy Parcel Service |
 * | rr_donnelley_ipa | International Priority Airmail (IPA)|
 * | rr_donnelley_courier | International Courier|
 * | rr_donnelley_isal | International Surface Air Lift (ISAL)|
 * | rr_donnelley_epacket | e-Packet|
 * | rr_donnelley_pmi | Priority Mail International|
 * | rr_donnelley_emi | Express Mail International|
 *
 */
exports.ServiceLevelePostGlobalEnum = {
    RrDonnelleyDomesticEconomyParcel: "rr_donnelley_domestic_economy_parcel",
    RrDonnelleyDomesticPriorityParcel: "rr_donnelley_domestic_priority_parcel",
    RrDonnelleyDomesticParcelBpm: "rr_donnelley_domestic_parcel_bpm",
    RrDonnelleyPriorityDomesticPriorityParcelBpm: "rr_donnelley_priority_domestic_priority_parcel_bpm",
    RrDonnelleyPriorityParcelDelcon: "rr_donnelley_priority_parcel_delcon",
    RrDonnelleyPriorityParcelNondelcon: "rr_donnelley_priority_parcel_nondelcon",
    RrDonnelleyEconomyParcel: "rr_donnelley_economy_parcel",
    RrDonnelleyIpa: "rr_donnelley_ipa",
    RrDonnelleyCourier: "rr_donnelley_courier",
    RrDonnelleyIsal: "rr_donnelley_isal",
    RrDonnelleyEpacket: "rr_donnelley_epacket",
    RrDonnelleyPmi: "rr_donnelley_pmi",
    RrDonnelleyEmi: "rr_donnelley_emi",
};
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | poste_italiane_delivery_business_express | Poste Delivery Business Express|
 *
 */
exports.ServiceLevelPostItalianeEnum = {
    PosteItalianeDeliveryBusinessExpress: "poste_italiane_delivery_business_express",
};
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | parcelforce_express48 | Express 48|
 * | parcelforce_express24 | Express 24|
 * | parcelforce_expressam | Express AM|
 *
 */
exports.ServiceLevelParcelforceEnum = {
    ParcelforceExpress48: "parcelforce_express48",
    ParcelforceExpress24: "parcelforce_express24",
    ParcelforceExpressam: "parcelforce_expressam",
};
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | x_delivery_expedited | Expedited|
 *
 */
exports.ServiceLevelMaergoEnum = {
    XDeliveryExpedited: "x_delivery_expedited",
};
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | mondial_relay_pointrelais | Point Relais|
 *
 */
exports.ServiceLevelMondialRelayEnum = {
    MondialRelayPointrelais: "mondial_relay_pointrelais",
};
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | lso_ground | Ground|
 * | lso_economy_next_day | Economy Next Day|
 * | lso_saturday_delivery | Saturday Delivery|
 * | lso_2nd_day | 2nd Day|
 * | lso_priority_next_day | Priority Next Day|
 * | lso_early_overnight | Early Overnight|
 *
 */
exports.ServiceLevelLSOEnum = {
    LsoGround: "lso_ground",
    LsoEconomyNextDay: "lso_economy_next_day",
    LsoSaturdayDelivery: "lso_saturday_delivery",
    Lso2ndDay: "lso_2nd_day",
    LsoPriorityNextDay: "lso_priority_next_day",
    LsoEarlyOvernight: "lso_early_overnight",
};
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | gls_us_cps | GSO Ground|
 * | gls_us_eps | Early Priority Overnight|
 * | gls_us_ess | Early Saturday Delivery|
 * | gls_us_nps | Noon Priority Overnight|
 * | gls_us_pds | Priority Overnight|
 * | gls_us_sds | Saturday Delivery|
 *
 */
exports.ServiceLevelGLSUSEnum = {
    GlsUsCps: "gls_us_cps",
    GlsUsEps: "gls_us_eps",
    GlsUsEss: "gls_us_ess",
    GlsUsNps: "gls_us_nps",
    GlsUsPds: "gls_us_pds",
    GlsUsSds: "gls_us_sds",
};
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | globegistics_priority_mail_express_international | Globegistics Priority Mail Express International|
 * | globegistics_priority_mail_international | Globegistics Priority Mail International|
 * | globegistics_priority_mail_express_international_pds | Globegistics Priority Mail Express International PreSort Drop Ship|
 * | globegistics_priority_mail_international_pds | Globegistics Priority Mail International PreSort Drop Ship|
 * | globegistics_epacket | Globegistics ePacket|
 * | globegistics_ecom_tracked_ddp | Globegistics eCom Tracked DDP|
 * | globegistics_ecom_packet_ddp | Globegistics eCom Packet DDP|
 * | globegistics_ecom_priority_mail_international_ddp | Globegistics eCom Priority Mail International DDP|
 * | globegistics_ecom_priority_mail_express_international_ddp | Globegistics eCom Priority Mail Express International DDP|
 * | globegistics_ecom_extra | Globegistics eCom Extra|
 * | globegistics_ecom_international_priority_airmail | Globegistics eCom International Priority Airmail|
 * | globegistics_ecom_international_surface_airlift | Globegistics eCom International Surface Air Lift|
 *
 */
exports.ServiceLevelGlobegisticsEnum = {
    GlobegisticsPriorityMailExpressInternational: "globegistics_priority_mail_express_international",
    GlobegisticsPriorityMailInternational: "globegistics_priority_mail_international",
    GlobegisticsPriorityMailExpressInternationalPds: "globegistics_priority_mail_express_international_pds",
    GlobegisticsPriorityMailInternationalPds: "globegistics_priority_mail_international_pds",
    GlobegisticsEpacket: "globegistics_epacket",
    GlobegisticsEcomTrackedDdp: "globegistics_ecom_tracked_ddp",
    GlobegisticsEcomPacketDdp: "globegistics_ecom_packet_ddp",
    GlobegisticsEcomPriorityMailInternationalDdp: "globegistics_ecom_priority_mail_international_ddp",
    GlobegisticsEcomPriorityMailExpressInternationalDdp: "globegistics_ecom_priority_mail_express_international_ddp",
    GlobegisticsEcomExtra: "globegistics_ecom_extra",
    GlobegisticsEcomInternationalPriorityAirmail: "globegistics_ecom_international_priority_airmail",
    GlobegisticsEcomInternationalSurfaceAirlift: "globegistics_ecom_international_surface_airlift",
};
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | fastway_australia_parcel | Parcel|
 * | fastway_australia_satchel | Satchel|
 * | fastway_australia_box_small | Box Small|
 * | fastway_australia_box_medium | Box Medium|
 * | fastway_australia_box_large | Box Large|
 *
 */
exports.ServiceLevelAramexAustraliaEnum = {
    FastwayAustraliaParcel: "fastway_australia_parcel",
    FastwayAustraliaSatchel: "fastway_australia_satchel",
    FastwayAustraliaBoxSmall: "fastway_australia_box_small",
    FastwayAustraliaBoxMedium: "fastway_australia_box_medium",
    FastwayAustraliaBoxLarge: "fastway_australia_box_large",
};
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | deutsche_post_postkarte | Postkarte|
 * | deutsche_post_standardbrief | Standardbrief|
 * | deutsche_post_kompaktbrief | Kompaktbrief|
 * | deutsche_post_grossbrief | Grossbrief|
 * | deutsche_post_maxibrief | Maxibrief|
 * | deutsche_post_maxibrief_plus | Maxibrief Plus|
 * | deutsche_post_warenpost_international_xs | Warenpost International XS|
 * | deutsche_post_warenpost_international_s | Warenpost International S|
 * | deutsche_post_warenpost_international_m | Warenpost International M|
 * | deutsche_post_warenpost_international_l | Warenpost International L|
 *
 */
exports.ServiceLevelDeutschePostEnum = {
    DeutschePostPostkarte: "deutsche_post_postkarte",
    DeutschePostStandardbrief: "deutsche_post_standardbrief",
    DeutschePostKompaktbrief: "deutsche_post_kompaktbrief",
    DeutschePostGrossbrief: "deutsche_post_grossbrief",
    DeutschePostMaxibrief: "deutsche_post_maxibrief",
    DeutschePostMaxibriefPlus: "deutsche_post_maxibrief_plus",
    DeutschePostWarenpostInternationalXs: "deutsche_post_warenpost_international_xs",
    DeutschePostWarenpostInternationalS: "deutsche_post_warenpost_international_s",
    DeutschePostWarenpostInternationalM: "deutsche_post_warenpost_international_m",
    DeutschePostWarenpostInternationalL: "deutsche_post_warenpost_international_l",
};
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | dpd_uk_ship_to_shop | Ship to shop|
 * | dpd_uk_1030 | Door to door 10.30 next day|
 * | dpd_uk_1200 | Door to door 12.00 next day|
 * | dpd_uk_saturday | Saturday Delivery|
 * | dpd_uk_saturday_1030 | Saturday Delivery 10.30|
 * | dpd_uk_saturday_1200 | Saturday Delivery 12.00|
 * | dpd_uk_sunday | Sunday Delivery|
 * | dpd_uk_sunday_1030 | Sunday Delivery 10.30|
 * | dpd_uk_sunday_1200 | Sunday Delivery 12.00|
 * | dpd_uk_next_day | Next day|
 * | dpd_uk_two_day | Two day|
 * | dpd_uk_classic | DPD Classic|
 * | dpd_uk_air_classic | DPD Air Classic|
 * | dpd_uk_air_express | DPD Air Express|
 * | dpd_uk_direct | DPD Direct|
 * | dpd_uk_direct_tracked_mail| DPD Tracked Mail|
 * | dpd_uk_pickup_returns | Pickup Returns|
 *
 */
exports.ServiceLevelDPDUKEnum = {
    DpdUkShipToShop: "dpd_uk_ship_to_shop",
    DpdUk1030: "dpd_uk_1030",
    DpdUk1200: "dpd_uk_1200",
    DpdUkSaturday: "dpd_uk_saturday",
    DpdUkSaturday1030: "dpd_uk_saturday_1030",
    DpdUkSaturday1200: "dpd_uk_saturday_1200",
    DpdUkSunday: "dpd_uk_sunday",
    DpdUkSunday1030: "dpd_uk_sunday_1030",
    DpdUkSunday1200: "dpd_uk_sunday_1200",
    DpdUkNextDay: "dpd_uk_next_day",
    DpdUkTwoDay: "dpd_uk_two_day",
    DpdUkClassic: "dpd_uk_classic",
    DpdUkAirClassic: "dpd_uk_air_classic",
    DpdUkAirExpress: "dpd_uk_air_express",
    DpdUkDirect: "dpd_uk_direct",
    DpdUkDirectTrackedMail: "dpd_uk_direct_tracked_mail",
    DpdUkPickupReturns: "dpd_uk_pickup_returns",
};
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | dpd_de_classic | DPD Classic|
 *
 */
exports.ServiceLevelDPDDEEnum = {
    DpdDeClassic: "dpd_de_classic",
};
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | dhl_germany_europaket | DHL Germany Europaket|
 * | dhl_germany_paket | DHL Germany Paket|
 * | dhl_germany_paket_connect | DHL Germany Paket Connect|
 * | dhl_germany_paket_international | DHL Germany Paket International|
 * | dhl_germany_paket_priority | DHL Germany Paket Priority|
 * | dhl_germany_paket_sameday | DHL Germany Paket Sameday|
 *
 */
exports.ServiceLevelDHLGermanyEnum = {
    DhlGermanyEuropaket: "dhl_germany_europaket",
    DhlGermanyPaket: "dhl_germany_paket",
    DhlGermanyPaketConnect: "dhl_germany_paket_connect",
    DhlGermanyPaketInternational: "dhl_germany_paket_international",
    DhlGermanyPaketPriority: "dhl_germany_paket_priority",
    DhlGermanyPaketSameday: "dhl_germany_paket_sameday",
};
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | dhl_ecommerce_marketing_parcel_expedited | Marketing Parcel Expedited|
 * | dhl_ecommerce_globalmail_business_ipa | GlobalMail Business IPA|
 * | dhl_ecommerce_parcel_international_direct | Parcel International Direct|
 * | dhl_ecommerce_parcels_expedited_max | Parcels Expedited Max|
 * | dhl_ecommerce_bpm_ground | Bounded Printed Matter Ground|
 * | dhl_ecommerce_priority_expedited | Priority Expedited|
 * | dhl_ecommerce_globalmail_packet_ipa | GlobalMail Packet IPA|
 * | dhl_ecommerce_globalmail_packet_isal | GlobalMail Packet ISAL|
 * | dhl_ecommerce_easy_return_plus | Easy Return Plus|
 * | dhl_ecommerce_marketing_parcel_ground | Marketing Parcel Ground|
 * | dhl_ecommerce_first_class_parcel_expedited | First Class Parcel Expedited|
 * | dhl_ecommerce_globalmail_business_priority | GlobalMail Business Priority|
 * | dhl_ecommerce_parcels_expedited | Parcels Expedited|
 * | dhl_ecommerce_globalmail_business_isal | GlobalMail Business ISAL|
 * | dhl_ecommerce_parcel_plus_expedited_max | Parcel Plus Expedited Max|
 * | dhl_ecommerce_globalmail_packet_plus | GlobalMail Packet Plus|
 * | dhl_ecommerce_parcels_ground | Parcels Ground|
 * | dhl_ecommerce_expedited | Expedited|
 * | dhl_ecommerce_parcel_plus_ground | Parcel Plus Ground|
 * | dhl_ecommerce_parcel_international_standard | Parcel International Standard|
 * | dhl_ecommerce_bpm_expedited | Bounded Printed Matter Expedited|
 * | dhl_ecommerce_parcel_international_expedited | Parcel International Expedited|
 * | dhl_ecommerce_globalmail_packet_priority | GlobalMail Packet Priority|
 * | dhl_ecommerce_easy_return_light | Easy Return Light|
 * | dhl_ecommerce_parcel_plus_expedited | Parcel Plus Expedited|
 * | dhl_ecommerce_globalmail_business_standard | GlobalMail Packet Standard|
 * | dhl_ecommerce_ground | Ground|
 * | dhl_ecommerce_globalmail_packet_standard | GlobalMail Business Standard|
 *
 */
exports.ServiceLevelDHLeCommerceEnum = {
    DhlEcommerceMarketingParcelExpedited: "dhl_ecommerce_marketing_parcel_expedited",
    DhlEcommerceGlobalmailBusinessIpa: "dhl_ecommerce_globalmail_business_ipa",
    DhlEcommerceParcelInternationalDirect: "dhl_ecommerce_parcel_international_direct",
    DhlEcommerceParcelsExpeditedMax: "dhl_ecommerce_parcels_expedited_max",
    DhlEcommerceBpmGround: "dhl_ecommerce_bpm_ground",
    DhlEcommercePriorityExpedited: "dhl_ecommerce_priority_expedited",
    DhlEcommerceGlobalmailPacketIpa: "dhl_ecommerce_globalmail_packet_ipa",
    DhlEcommerceGlobalmailPacketIsal: "dhl_ecommerce_globalmail_packet_isal",
    DhlEcommerceEasyReturnPlus: "dhl_ecommerce_easy_return_plus",
    DhlEcommerceMarketingParcelGround: "dhl_ecommerce_marketing_parcel_ground",
    DhlEcommerceFirstClassParcelExpedited: "dhl_ecommerce_first_class_parcel_expedited",
    DhlEcommerceGlobalmailBusinessPriority: "dhl_ecommerce_globalmail_business_priority",
    DhlEcommerceParcelsExpedited: "dhl_ecommerce_parcels_expedited",
    DhlEcommerceGlobalmailBusinessIsal: "dhl_ecommerce_globalmail_business_isal",
    DhlEcommerceParcelPlusExpeditedMax: "dhl_ecommerce_parcel_plus_expedited_max",
    DhlEcommerceGlobalmailPacketPlus: "dhl_ecommerce_globalmail_packet_plus",
    DhlEcommerceParcelsGround: "dhl_ecommerce_parcels_ground",
    DhlEcommerceExpedited: "dhl_ecommerce_expedited",
    DhlEcommerceParcelPlusGround: "dhl_ecommerce_parcel_plus_ground",
    DhlEcommerceParcelInternationalStandard: "dhl_ecommerce_parcel_international_standard",
    DhlEcommerceBpmExpedited: "dhl_ecommerce_bpm_expedited",
    DhlEcommerceParcelInternationalExpedited: "dhl_ecommerce_parcel_international_expedited",
    DhlEcommerceGlobalmailPacketPriority: "dhl_ecommerce_globalmail_packet_priority",
    DhlEcommerceEasyReturnLight: "dhl_ecommerce_easy_return_light",
    DhlEcommerceParcelPlusExpedited: "dhl_ecommerce_parcel_plus_expedited",
    DhlEcommerceGlobalmailBusinessStandard: "dhl_ecommerce_globalmail_business_standard",
    DhlEcommerceGround: "dhl_ecommerce_ground",
    DhlEcommerceGlobalmailPacketStandard: "dhl_ecommerce_globalmail_packet_standard",
};
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | dhl_express_domestic_express_doc | Domestic Express Doc|
 * | dhl_express_economy_select_doc | Economy Select Doc|
 * | dhl_express_worldwide_nondoc | Express Worldwide Nondoc|
 * | dhl_express_worldwide_doc | Express Worldwide Doc|
 * | dhl_express_worldwide | Worldwide|
 * | dhl_express_worldwide_eu_doc | Express Worldwide EU Doc|
 * | dhl_express_break_bulk_express_doc | Break Bulk Express Doc|
 * | dhl_express_express_9_00_nondoc | Express 9:00 NonDoc|
 * | dhl_express_economy_select_nondoc | Economy Select NonDoc|
 * | dhl_express_break_bulk_economy_doc | Break Bulk Economy Doc|
 * | dhl_express_express_9_00_doc | Express 9:00 Doc|
 * | dhl_express_express_10_30_doc | Express 10:30 Doc|
 * | dhl_express_express_10_30_nondoc | Express 10:30 NonDoc|
 * | dhl_express_express_12_00_doc | Express 12:00 Doc|
 * | dhl_express_europack_nondoc | Europack NonDoc|
 * | dhl_express_express_envelope_doc | Express Envelope Doc|
 * | dhl_express_express_12_00_nondoc | Express 12:00 NonDoc|
 * | dhl_express_express_12_doc | Domestic Express 12:00|
 * | dhl_express_worldwide_b2c_doc | Express Worldwide (B2C) Doc|
 * | dhl_express_worldwide_b2c_nondoc | Express Worldwide (B2C) NonDoc|
 * | dhl_express_medical_express | Medical Express|
 * | dhl_express_express_easy_nondoc | Express Easy NonDoc|
 *
 */
exports.ServiceLevelDHLExpressEnum = {
    DhlExpressDomesticExpressDoc: "dhl_express_domestic_express_doc",
    DhlExpressEconomySelectDoc: "dhl_express_economy_select_doc",
    DhlExpressWorldwideNondoc: "dhl_express_worldwide_nondoc",
    DhlExpressWorldwideDoc: "dhl_express_worldwide_doc",
    DhlExpressWorldwide: "dhl_express_worldwide",
    DhlExpressWorldwideEuDoc: "dhl_express_worldwide_eu_doc",
    DhlExpressBreakBulkExpressDoc: "dhl_express_break_bulk_express_doc",
    DhlExpressExpress900Nondoc: "dhl_express_express_9_00_nondoc",
    DhlExpressEconomySelectNondoc: "dhl_express_economy_select_nondoc",
    DhlExpressBreakBulkEconomyDoc: "dhl_express_break_bulk_economy_doc",
    DhlExpressExpress900Doc: "dhl_express_express_9_00_doc",
    DhlExpressExpress1030Doc: "dhl_express_express_10_30_doc",
    DhlExpressExpress1030Nondoc: "dhl_express_express_10_30_nondoc",
    DhlExpressExpress1200Doc: "dhl_express_express_12_00_doc",
    DhlExpressEuropackNondoc: "dhl_express_europack_nondoc",
    DhlExpressExpressEnvelopeDoc: "dhl_express_express_envelope_doc",
    DhlExpressExpress1200Nondoc: "dhl_express_express_12_00_nondoc",
    DhlExpressExpress12Doc: "dhl_express_express_12_doc",
    DhlExpressWorldwideB2cDoc: "dhl_express_worldwide_b2c_doc",
    DhlExpressWorldwideB2cNondoc: "dhl_express_worldwide_b2c_nondoc",
    DhlExpressMedicalExpress: "dhl_express_medical_express",
    DhlExpressExpressEasyNondoc: "dhl_express_express_easy_nondoc",
};
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | purolator_ground | Ground|
 * | purolator_ground9_am | Ground 9am|
 * | purolator_ground1030_am | Ground 10:30am|
 * | purolator_ground_distribution | Ground Distribution|
 * | purolator_ground_evening | Ground Evening|
 * | purolator_ground_us | Ground US|
 * | purolator_express | Express|
 * | purolator_express9_am | Express 9am|
 * | purolator_express1030_am | Express 10am|
 * | purolator_express_evening | Express Evening|
 * | purolator_express_us | Express US|
 * | purolator_express_us9_am | Express US 9am|
 * | purolator_express_us1030_am | Express US 10:30am|
 * | purolator_express_us1200 | Express US 12pm|
 * | purolator_express_international | Express International|
 * | purolator_express_international9_am | Express International 9am|
 * | purolator_express_international1030_am | Express International 10:30am|
 * | purolator_express_international1200 | Express International 12pm|
 *
 */
exports.ServiceLevelPurolatorEnum = {
    PurolatorGround: "purolator_ground",
    PurolatorGround9Am: "purolator_ground9_am",
    PurolatorGround1030Am: "purolator_ground1030_am",
    PurolatorGroundDistribution: "purolator_ground_distribution",
    PurolatorGroundEvening: "purolator_ground_evening",
    PurolatorGroundUs: "purolator_ground_us",
    PurolatorExpress: "purolator_express",
    PurolatorExpress9Am: "purolator_express9_am",
    PurolatorExpress1030Am: "purolator_express1030_am",
    PurolatorExpressEvening: "purolator_express_evening",
    PurolatorExpressUs: "purolator_express_us",
    PurolatorExpressUs9Am: "purolator_express_us9_am",
    PurolatorExpressUs1030Am: "purolator_express_us1030_am",
    PurolatorExpressUs1200: "purolator_express_us1200",
    PurolatorExpressInternational: "purolator_express_international",
    PurolatorExpressInternational9Am: "purolator_express_international9_am",
    PurolatorExpressInternational1030Am: "purolator_express_international1030_am",
    PurolatorExpressInternational1200: "purolator_express_international1200",
};
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | colissimo_home | Domicile|
 * | colissimo_pick_up_point | Point Retrait|
 * | colissimo_return_mainland_france | Retour France|
 *
 */
exports.ServiceLevelColissimoEnum = {
    ColissimoHome: "colissimo_home",
    ColissimoPickUpPoint: "colissimo_pick_up_point",
    ColissimoReturnMainlandFrance: "colissimo_return_mainland_france",
};
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | correos_standard_home| Paquete Estándar|
 * | correos_premium_home | Paquete Premium|
 *
 */
exports.ServiceLevelCorreosEspanaEnum = {
    CorreosStandardHome: "correos_standard_home",
    CorreosPremiumHome: "correos_premium_home",
};
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | couriersplease_domestic_priority_auth_to_leave | Domestic Priority - Authority To Leave/POPPoints|
 * | couriersplease_domestic_priority_sign_required | Domestic Priority - Signature Required|
 * | couriersplease_gold_domestic_auth_to_leave | Gold Domestic - Authority To Leave/POPPoints|
 * | couriersplease_gold_domestic_sign_required | Gold Domestic - Signature Required|
 * | couriersplease_off_peak_auth_to_leave | Off Peak - Authority To Leave/POPPoints|
 * | couriersplease_off_peak_sign_required | Off Peak - Signature Required|
 * | couriersplease_parcel_auth_to_leave | Parcel - Authority To Leave|
 * | couriersplease_parcel_sign_required | Parcel - Signature Required|
 * | couriersplease_road_express | Road Express|
 * | couriersplease_satchel_auth_to_leave | Satchel - Authority To Leave|
 * | couriersplease_satchel_sign_required | Satchel - Signature Required|
 *
 */
exports.ServiceLevelCouriersPleaseEnum = {
    CourierspleaseDomesticPriorityAuthToLeave: "couriersplease_domestic_priority_auth_to_leave",
    CourierspleaseDomesticPrioritySignRequired: "couriersplease_domestic_priority_sign_required",
    CourierspleaseGoldDomesticAuthToLeave: "couriersplease_gold_domestic_auth_to_leave",
    CourierspleaseGoldDomesticSignRequired: "couriersplease_gold_domestic_sign_required",
    CourierspleaseOffPeakAuthToLeave: "couriersplease_off_peak_auth_to_leave",
    CourierspleaseOffPeakSignRequired: "couriersplease_off_peak_sign_required",
    CourierspleaseParcelAuthToLeave: "couriersplease_parcel_auth_to_leave",
    CourierspleaseParcelSignRequired: "couriersplease_parcel_sign_required",
    CourierspleaseRoadExpress: "couriersplease_road_express",
    CourierspleaseSatchelAuthToLeave: "couriersplease_satchel_auth_to_leave",
    CourierspleaseSatchelSignRequired: "couriersplease_satchel_sign_required",
};
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | chronopost_13 | Chrono 13|
 * | chronopost_10 | Chrono 10|
 * | chronopost_18| Chrono 18|
 * | chronopost_relais_fr | Chrono Point Relais|
 * | chronopost_classic | Chrono Classic International|
 * | chronopost_express | Chrono Express International|
 *
 */
exports.ServiceLevelChronopostEnum = {
    Chronopost13: "chronopost_13",
    Chronopost10: "chronopost_10",
    Chronopost18: "chronopost_18",
    ChronopostRelaisFr: "chronopost_relais_fr",
    ChronopostClassic: "chronopost_classic",
    ChronopostExpress: "chronopost_express",
};
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | cdl_next_day | Next Day|
 *
 */
exports.ServiceLevelCDLEnum = {
    CdlNextDay: "cdl_next_day",
};
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | canada_post_regular_parcel | Regular Parcel|
 * | canada_post_expedited_parcel | Expedited Parcel|
 * | canada_post_priority | Priority|
 * | canada_post_xpresspost | Xpresspost|
 * | canada_post_xpresspost_international | Xpresspost International|
 * | canada_post_xpresspost_usa | Xpresspost USA|
 * | canada_post_expedited_parcel_usa | Expedited Parcel USA|
 * | canada_post_tracked_packet_usa | Tracked Packet USA|
 * | canada_post_small_packet_usa_air | Small Packet USA Air|
 * | canada_post_tracked_packet_international | Tracked Packet International|
 * | canada_post_small_packet_international_air | Small Package International Air|
 *
 */
exports.ServiceLevelCanadaPostEnum = {
    CanadaPostRegularParcel: "canada_post_regular_parcel",
    CanadaPostExpeditedParcel: "canada_post_expedited_parcel",
    CanadaPostPriority: "canada_post_priority",
    CanadaPostXpresspost: "canada_post_xpresspost",
    CanadaPostXpresspostInternational: "canada_post_xpresspost_international",
    CanadaPostXpresspostUsa: "canada_post_xpresspost_usa",
    CanadaPostExpeditedParcelUsa: "canada_post_expedited_parcel_usa",
    CanadaPostTrackedPacketUsa: "canada_post_tracked_packet_usa",
    CanadaPostSmallPacketUsaAir: "canada_post_small_packet_usa_air",
    CanadaPostTrackedPacketInternational: "canada_post_tracked_packet_international",
    CanadaPostSmallPacketInternationalAir: "canada_post_small_packet_international_air",
};
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | better_trucks_same_day | Same Day|
 * | better_trucks_next_day | Next Day|
 *
 */
exports.ServiceLevelBetterTrucksEnum = {
    BetterTrucksSameDay: "better_trucks_same_day",
    BetterTrucksNextDay: "better_trucks_next_day",
};
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | epacket | ePacket|
 * | epacket_plus | ePacket Plus|
 * | eparcel_premium | eParcel premium|
 * | apg_eparcel_expedited | eParcel Expedited|
 *
 */
exports.ServiceLevelAPGEnum = {
    Epacket: "epacket",
    EpacketPlus: "epacket_plus",
    EparcelPremium: "eparcel_premium",
    ApgEparcelExpedited: "apg_eparcel_expedited",
};
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | australia_post_express_post | Express Post|
 * | australia_post_parcel_post | Parcel Post|
 * | australia_post_pack_and_track_international | Pack and Track International|
 * | australia_post_international_airmail | International Airmail|
 * | australia_post_express_post_international | Express Post International|
 * | australia_post_express_courier_international | Express Courier International|
 * | australia_post_international_express | International Express|
 * | australia_post_international_standard | International Standard|
 * | australia_post_international_economy | International Economy|
 * | australia_post_parcel_post_return | Parcel Post Return|
 *
 */
exports.ServiceLevelAustraliaPostEnum = {
    AustraliaPostExpressPost: "australia_post_express_post",
    AustraliaPostParcelPost: "australia_post_parcel_post",
    AustraliaPostPackAndTrackInternational: "australia_post_pack_and_track_international",
    AustraliaPostInternationalAirmail: "australia_post_international_airmail",
    AustraliaPostExpressPostInternational: "australia_post_express_post_international",
    AustraliaPostExpressCourierInternational: "australia_post_express_courier_international",
    AustraliaPostInternationalExpress: "australia_post_international_express",
    AustraliaPostInternationalStandard: "australia_post_international_standard",
    AustraliaPostInternationalEconomy: "australia_post_international_economy",
    AustraliaPostParcelPostReturn: "australia_post_parcel_post_return",
};
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | asendia_us_priority_tracked | Asendia USA Priority Tracked|
 * | asendia_us_international_express | Asendia USA International Express|
 * | asendia_us_international_priority_airmail | Asendia USA International Priority Airmail|
 * | asendia_us_international_surface_airlift | Asendia USA International Surface Air Lift|
 * | asendia_us_priority_mail_international | Asendia USA Priority Mail International|
 * | asendia_us_priority_mail_express_international | Asendia USA Priority Mail Express International|
 * | asendia_us_epacket | Asendia USA International ePacket|
 * | asendia_us_other | Asendia USA Other Services (custom)|
 *
 */
exports.ServiceLevelAsendiaEnum = {
    AsendiaUsPriorityTracked: "asendia_us_priority_tracked",
    AsendiaUsInternationalExpress: "asendia_us_international_express",
    AsendiaUsInternationalPriorityAirmail: "asendia_us_international_priority_airmail",
    AsendiaUsInternationalSurfaceAirlift: "asendia_us_international_surface_airlift",
    AsendiaUsPriorityMailInternational: "asendia_us_priority_mail_international",
    AsendiaUsPriorityMailExpressInternational: "asendia_us_priority_mail_express_international",
    AsendiaUsEpacket: "asendia_us_epacket",
    AsendiaUsOther: "asendia_us_other",
};
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | apc_postal_parcelconnect_expedited | parcelConnect Expedited|
 * | apc_postal_parcelconnect_priority | parcelConnect Priority|
 * | apc_postal_parcelconnect_priority_delcon | parcelConnect Priority Delcon|
 * | apc_postal_parcelconnect_priority_pqw | parcelConnect Priority PQW|
 * | apc_postal_parcelconnect_book_service | parcelConnect Book Service|
 * | apc_postal_parcelconnect_standard | parcelConnect Standard|
 * | apc_postal_parcelconnect_epmi | parcelConnect ePMI|
 * | apc_postal_parcelconnect_epacket | parcelConnect ePacket|
 * | apc_postal_parcelconnect_epmei | parcelConnect ePMEI|
 *
 */
exports.ServiceLevelAPCPostalEnum = {
    ApcPostalParcelconnectExpedited: "apc_postal_parcelconnect_expedited",
    ApcPostalParcelconnectPriority: "apc_postal_parcelconnect_priority",
    ApcPostalParcelconnectPriorityDelcon: "apc_postal_parcelconnect_priority_delcon",
    ApcPostalParcelconnectPriorityPqw: "apc_postal_parcelconnect_priority_pqw",
    ApcPostalParcelconnectBookService: "apc_postal_parcelconnect_book_service",
    ApcPostalParcelconnectStandard: "apc_postal_parcelconnect_standard",
    ApcPostalParcelconnectEpmi: "apc_postal_parcelconnect_epmi",
    ApcPostalParcelconnectEpacket: "apc_postal_parcelconnect_epacket",
    ApcPostalParcelconnectEpmei: "apc_postal_parcelconnect_epmei",
};
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | airterra_ground | GroundXC|
 * | airterra_postal | FastPost|
 *
 */
exports.ServiceLevelAirterraEnum = {
    AirterraGround: "airterra_ground",
    AirterraPostal: "airterra_postal",
};
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | ups_standard | Standard℠|
 * | ups_ground | Ground|
 * | ups_saver | Saver®|
 * | ups_3_day_select | 3 Day Select®|
 * | ups_second_day_air | 2nd Day Air®|
 * | ups_second_day_air_am | 2nd Day Air® A.M.|
 * | ups_next_day_air | Next Day Air®|
 * | ups_next_day_air_saver | Next Day Air Saver®|
 * | ups_next_day_air_early_am | Next Day Air® Early|
 * | ups_mail_innovations_bpm_flat | Mail Innovations BPM Flat |
 * | ups_mail_innovations_bpm_parcel | Mail Innovations BPM Parcel |
 * | ups_mail_innovations_irregular | Mail Innovations Irregular |
 * | ups_mail_innovations_media_mail | Mail Innovations Media Mail |
 * | ups_mail_innovations_parcel | Mail Innovations Parcel |
 * | ups_mail_innovations_standard_flat | Mail Innovations Standard Flat |
 * | ups_surepost | Surepost|
 * | ups_surepost_bound_printed_matter | SurePost® Bound Printed Matter|
 * | ups_surepost_lightweight | Surepost Lightweight|
 * | ups_surepost_media | SurePost® Media|
 * | ups_express | Express®|
 * | ups_express_1200 | Express 12:00|
 * | ups_express_plus | Express Plus®|
 * | ups_expedited | Expedited®|
 * | ups_express_early | Express® Early|
 * | ups_access_point_economy | Access Point™ Economy|
 *
 */
exports.ServiceLevelUPSEnum = {
    UpsStandard: "ups_standard",
    UpsGround: "ups_ground",
    UpsSaver: "ups_saver",
    Ups3DaySelect: "ups_3_day_select",
    UpsSecondDayAir: "ups_second_day_air",
    UpsSecondDayAirAm: "ups_second_day_air_am",
    UpsNextDayAir: "ups_next_day_air",
    UpsNextDayAirSaver: "ups_next_day_air_saver",
    UpsNextDayAirEarlyAm: "ups_next_day_air_early_am",
    UpsMailInnovationsBpmFlat: "ups_mail_innovations_bpm_flat",
    UpsMailInnovationsBpmParcel: "ups_mail_innovations_bpm_parcel",
    UpsMailInnovationsIrregular: "ups_mail_innovations_irregular",
    UpsMailInnovationsMediaMail: "ups_mail_innovations_media_mail",
    UpsMailInnovationsParcel: "ups_mail_innovations_parcel",
    UpsMailInnovationsStandardFlat: "ups_mail_innovations_standard_flat",
    UpsSurepost: "ups_surepost",
    UpsSurepostBoundPrintedMatter: "ups_surepost_bound_printed_matter",
    UpsSurepostLightweight: "ups_surepost_lightweight",
    UpsSurepostMedia: "ups_surepost_media",
    UpsExpress: "ups_express",
    UpsExpress1200: "ups_express_1200",
    UpsExpressPlus: "ups_express_plus",
    UpsExpedited: "ups_expedited",
    UpsExpressEarly: "ups_express_early",
    UpsAccessPointEconomy: "ups_access_point_economy",
};
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | fedex_ground | FedEx Ground®|
 * | fedex_home_delivery | FedEx Home Delivery®|
 * | fedex_ground_economy | FedEx Ground® Economy|
 * | fedex_2_day | FedEx 2Day®|
 * | fedex_2_day_am | FedEx 2Day® A.M.|
 * | fedex_express_saver | FedEx Express Saver®|
 * | fedex_standard_overnight | FedEx Standard Overnight®|
 * | fedex_priority_overnight | FedEx Priority Overnight®|
 * | fedex_first_overnight | FedEx First Overnight®|
 * | fedex_international_economy | FedEx International Economy®|
 * | fedex_international_priority | FedEx International Priority®|
 * | fedex_international_first | FedEx International First®|
 * | fedex_europe_first_international_priority | FedEx International First®|
 * | fedex_international_connect_plus | FedEx International Connect Plus|
 * | fedex_first | FedEx First |
 * | fedex_priority | FedEx Priority |
 * | fedex_priority_express | FedEx Priority Express|
 * | fedex_economy_select | FedEx® Economy |
 * | fedex_regional_economy | FedEx® Regional Economy |
 * | fedex_first_overnight_extra_hours | First Overnight® EH |
 * | fedex_international_priority_express | International Priority® Express |
 * | fedex_next_day_mid_morning | Next Day Mid Morning |
 * | fedex_priority_overnight_extra_hours | Priority Overnight® EH |
 * | fedex_standard_overnight_extra_hours | Standard Overnight® EH |
 *
 */
exports.ServiceLevelFedExEnum = {
    FedexGround: "fedex_ground",
    FedexHomeDelivery: "fedex_home_delivery",
    FedexSmartPost: "fedex_smart_post",
    FedexGroundEconomy: "fedex_ground_economy",
    Fedex2Day: "fedex_2_day",
    Fedex2DayAm: "fedex_2_day_am",
    FedexExpressSaver: "fedex_express_saver",
    FedexStandardOvernight: "fedex_standard_overnight",
    FedexPriorityOvernight: "fedex_priority_overnight",
    FedexFirstOvernight: "fedex_first_overnight",
    FedexInternationalEconomy: "fedex_international_economy",
    FedexInternationalPriority: "fedex_international_priority",
    FedexInternationalFirst: "fedex_international_first",
    FedexEuropeFirstInternationalPriority: "fedex_europe_first_international_priority",
    FedexInternationalConnectPlus: "fedex_international_connect_plus",
    FedexFirst: "fedex_first",
    FedexPriority: "fedex_priority",
    FedexPriorityExpress: "fedex_priority_express",
    FedexEconomySelect: "fedex_economy_select",
    FedexRegionalEconomy: "fedex_regional_economy",
    FedexFirstOvernightExtraHours: "fedex_first_overnight_extra_hours",
    FedexInternationalPriorityExpress: "fedex_international_priority_express",
    FedexNextDayMidMorning: "fedex_next_day_mid_morning",
    FedexPriorityOvernightExtraHours: "fedex_priority_overnight_extra_hours",
    FedexStandardOvernightExtraHours: "fedex_standard_overnight_extra_hours",
};
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | usps_priority | Priority Mail|
 * | usps_priority_express | Priority Mail Express|
 * | usps_media_mail | Media Mail, only for existing Shippo customers with grandfathered Media Mail option.|
 * | usps_priority_mail_international | Priority Mail International|
 * | usps_priority_mail_express_international | Priority Mail Express International|
 * | usps_first_class_package_international_service | First Class Package International|
 * | usps_ground_advantage | Ground Advantage |
 *
 */
exports.ServiceLevelUSPSEnum = {
    UspsPriority: "usps_priority",
    UspsPriorityExpress: "usps_priority_express",
    UspsMediaMail: "usps_media_mail",
    UspsPriorityMailInternational: "usps_priority_mail_international",
    UspsPriorityMailExpressInternational: "usps_priority_mail_express_international",
    UspsFirstClassPackageInternationalService: "usps_first_class_package_international_service",
    UspsGroundAdvantage: "usps_ground_advantage",
};
/** @internal */
var ServiceLevelSwyftEnum$;
(function (ServiceLevelSwyftEnum$) {
    ServiceLevelSwyftEnum$.inboundSchema = z.nativeEnum(exports.ServiceLevelSwyftEnum);
    ServiceLevelSwyftEnum$.outboundSchema = ServiceLevelSwyftEnum$.inboundSchema;
})(ServiceLevelSwyftEnum$ || (exports.ServiceLevelSwyftEnum$ = ServiceLevelSwyftEnum$ = {}));
/** @internal */
var ServiceLevelVehoEnum$;
(function (ServiceLevelVehoEnum$) {
    ServiceLevelVehoEnum$.inboundSchema = z.nativeEnum(exports.ServiceLevelVehoEnum);
    ServiceLevelVehoEnum$.outboundSchema = ServiceLevelVehoEnum$.inboundSchema;
})(ServiceLevelVehoEnum$ || (exports.ServiceLevelVehoEnum$ = ServiceLevelVehoEnum$ = {}));
/** @internal */
var ServiceLevelUDSEnum$;
(function (ServiceLevelUDSEnum$) {
    ServiceLevelUDSEnum$.inboundSchema = z.nativeEnum(exports.ServiceLevelUDSEnum);
    ServiceLevelUDSEnum$.outboundSchema = ServiceLevelUDSEnum$.inboundSchema;
})(ServiceLevelUDSEnum$ || (exports.ServiceLevelUDSEnum$ = ServiceLevelUDSEnum$ = {}));
/** @internal */
var ServiceLevelEvriUKEnum$;
(function (ServiceLevelEvriUKEnum$) {
    ServiceLevelEvriUKEnum$.inboundSchema = z.nativeEnum(exports.ServiceLevelEvriUKEnum);
    ServiceLevelEvriUKEnum$.outboundSchema = ServiceLevelEvriUKEnum$.inboundSchema;
})(ServiceLevelEvriUKEnum$ || (exports.ServiceLevelEvriUKEnum$ = ServiceLevelEvriUKEnum$ = {}));
/** @internal */
var ServiceLevelLasershipEnum$;
(function (ServiceLevelLasershipEnum$) {
    ServiceLevelLasershipEnum$.inboundSchema = z.nativeEnum(exports.ServiceLevelLasershipEnum);
    ServiceLevelLasershipEnum$.outboundSchema = ServiceLevelLasershipEnum$.inboundSchema;
})(ServiceLevelLasershipEnum$ || (exports.ServiceLevelLasershipEnum$ = ServiceLevelLasershipEnum$ = {}));
/** @internal */
var ServiceLevelJitsuEnum$;
(function (ServiceLevelJitsuEnum$) {
    ServiceLevelJitsuEnum$.inboundSchema = z.nativeEnum(exports.ServiceLevelJitsuEnum);
    ServiceLevelJitsuEnum$.outboundSchema = ServiceLevelJitsuEnum$.inboundSchema;
})(ServiceLevelJitsuEnum$ || (exports.ServiceLevelJitsuEnum$ = ServiceLevelJitsuEnum$ = {}));
/** @internal */
var ServiceLevelOnTracEnum$;
(function (ServiceLevelOnTracEnum$) {
    ServiceLevelOnTracEnum$.inboundSchema = z.nativeEnum(exports.ServiceLevelOnTracEnum);
    ServiceLevelOnTracEnum$.outboundSchema = ServiceLevelOnTracEnum$.inboundSchema;
})(ServiceLevelOnTracEnum$ || (exports.ServiceLevelOnTracEnum$ = ServiceLevelOnTracEnum$ = {}));
/** @internal */
var ServiceLevelSendleEnum$;
(function (ServiceLevelSendleEnum$) {
    ServiceLevelSendleEnum$.inboundSchema = z.nativeEnum(exports.ServiceLevelSendleEnum);
    ServiceLevelSendleEnum$.outboundSchema = ServiceLevelSendleEnum$.inboundSchema;
})(ServiceLevelSendleEnum$ || (exports.ServiceLevelSendleEnum$ = ServiceLevelSendleEnum$ = {}));
/** @internal */
var ServiceLevelRoyalMailEnum$;
(function (ServiceLevelRoyalMailEnum$) {
    ServiceLevelRoyalMailEnum$.inboundSchema = z.nativeEnum(exports.ServiceLevelRoyalMailEnum);
    ServiceLevelRoyalMailEnum$.outboundSchema = ServiceLevelRoyalMailEnum$.inboundSchema;
})(ServiceLevelRoyalMailEnum$ || (exports.ServiceLevelRoyalMailEnum$ = ServiceLevelRoyalMailEnum$ = {}));
/** @internal */
var ServiceLevelePostGlobalEnum$;
(function (ServiceLevelePostGlobalEnum$) {
    ServiceLevelePostGlobalEnum$.inboundSchema = z.nativeEnum(exports.ServiceLevelePostGlobalEnum);
    ServiceLevelePostGlobalEnum$.outboundSchema = ServiceLevelePostGlobalEnum$.inboundSchema;
})(ServiceLevelePostGlobalEnum$ || (exports.ServiceLevelePostGlobalEnum$ = ServiceLevelePostGlobalEnum$ = {}));
/** @internal */
var ServiceLevelPostItalianeEnum$;
(function (ServiceLevelPostItalianeEnum$) {
    ServiceLevelPostItalianeEnum$.inboundSchema = z.nativeEnum(exports.ServiceLevelPostItalianeEnum);
    ServiceLevelPostItalianeEnum$.outboundSchema = ServiceLevelPostItalianeEnum$.inboundSchema;
})(ServiceLevelPostItalianeEnum$ || (exports.ServiceLevelPostItalianeEnum$ = ServiceLevelPostItalianeEnum$ = {}));
/** @internal */
var ServiceLevelParcelforceEnum$;
(function (ServiceLevelParcelforceEnum$) {
    ServiceLevelParcelforceEnum$.inboundSchema = z.nativeEnum(exports.ServiceLevelParcelforceEnum);
    ServiceLevelParcelforceEnum$.outboundSchema = ServiceLevelParcelforceEnum$.inboundSchema;
})(ServiceLevelParcelforceEnum$ || (exports.ServiceLevelParcelforceEnum$ = ServiceLevelParcelforceEnum$ = {}));
/** @internal */
var ServiceLevelMaergoEnum$;
(function (ServiceLevelMaergoEnum$) {
    ServiceLevelMaergoEnum$.inboundSchema = z.nativeEnum(exports.ServiceLevelMaergoEnum);
    ServiceLevelMaergoEnum$.outboundSchema = ServiceLevelMaergoEnum$.inboundSchema;
})(ServiceLevelMaergoEnum$ || (exports.ServiceLevelMaergoEnum$ = ServiceLevelMaergoEnum$ = {}));
/** @internal */
var ServiceLevelMondialRelayEnum$;
(function (ServiceLevelMondialRelayEnum$) {
    ServiceLevelMondialRelayEnum$.inboundSchema = z.nativeEnum(exports.ServiceLevelMondialRelayEnum);
    ServiceLevelMondialRelayEnum$.outboundSchema = ServiceLevelMondialRelayEnum$.inboundSchema;
})(ServiceLevelMondialRelayEnum$ || (exports.ServiceLevelMondialRelayEnum$ = ServiceLevelMondialRelayEnum$ = {}));
/** @internal */
var ServiceLevelLSOEnum$;
(function (ServiceLevelLSOEnum$) {
    ServiceLevelLSOEnum$.inboundSchema = z.nativeEnum(exports.ServiceLevelLSOEnum);
    ServiceLevelLSOEnum$.outboundSchema = ServiceLevelLSOEnum$.inboundSchema;
})(ServiceLevelLSOEnum$ || (exports.ServiceLevelLSOEnum$ = ServiceLevelLSOEnum$ = {}));
/** @internal */
var ServiceLevelGLSUSEnum$;
(function (ServiceLevelGLSUSEnum$) {
    ServiceLevelGLSUSEnum$.inboundSchema = z.nativeEnum(exports.ServiceLevelGLSUSEnum);
    ServiceLevelGLSUSEnum$.outboundSchema = ServiceLevelGLSUSEnum$.inboundSchema;
})(ServiceLevelGLSUSEnum$ || (exports.ServiceLevelGLSUSEnum$ = ServiceLevelGLSUSEnum$ = {}));
/** @internal */
var ServiceLevelGlobegisticsEnum$;
(function (ServiceLevelGlobegisticsEnum$) {
    ServiceLevelGlobegisticsEnum$.inboundSchema = z.nativeEnum(exports.ServiceLevelGlobegisticsEnum);
    ServiceLevelGlobegisticsEnum$.outboundSchema = ServiceLevelGlobegisticsEnum$.inboundSchema;
})(ServiceLevelGlobegisticsEnum$ || (exports.ServiceLevelGlobegisticsEnum$ = ServiceLevelGlobegisticsEnum$ = {}));
/** @internal */
var ServiceLevelAramexAustraliaEnum$;
(function (ServiceLevelAramexAustraliaEnum$) {
    ServiceLevelAramexAustraliaEnum$.inboundSchema = z.nativeEnum(exports.ServiceLevelAramexAustraliaEnum);
    ServiceLevelAramexAustraliaEnum$.outboundSchema = ServiceLevelAramexAustraliaEnum$.inboundSchema;
})(ServiceLevelAramexAustraliaEnum$ || (exports.ServiceLevelAramexAustraliaEnum$ = ServiceLevelAramexAustraliaEnum$ = {}));
/** @internal */
var ServiceLevelDeutschePostEnum$;
(function (ServiceLevelDeutschePostEnum$) {
    ServiceLevelDeutschePostEnum$.inboundSchema = z.nativeEnum(exports.ServiceLevelDeutschePostEnum);
    ServiceLevelDeutschePostEnum$.outboundSchema = ServiceLevelDeutschePostEnum$.inboundSchema;
})(ServiceLevelDeutschePostEnum$ || (exports.ServiceLevelDeutschePostEnum$ = ServiceLevelDeutschePostEnum$ = {}));
/** @internal */
var ServiceLevelDPDUKEnum$;
(function (ServiceLevelDPDUKEnum$) {
    ServiceLevelDPDUKEnum$.inboundSchema = z.nativeEnum(exports.ServiceLevelDPDUKEnum);
    ServiceLevelDPDUKEnum$.outboundSchema = ServiceLevelDPDUKEnum$.inboundSchema;
})(ServiceLevelDPDUKEnum$ || (exports.ServiceLevelDPDUKEnum$ = ServiceLevelDPDUKEnum$ = {}));
/** @internal */
var ServiceLevelDPDDEEnum$;
(function (ServiceLevelDPDDEEnum$) {
    ServiceLevelDPDDEEnum$.inboundSchema = z.nativeEnum(exports.ServiceLevelDPDDEEnum);
    ServiceLevelDPDDEEnum$.outboundSchema = ServiceLevelDPDDEEnum$.inboundSchema;
})(ServiceLevelDPDDEEnum$ || (exports.ServiceLevelDPDDEEnum$ = ServiceLevelDPDDEEnum$ = {}));
/** @internal */
var ServiceLevelDHLGermanyEnum$;
(function (ServiceLevelDHLGermanyEnum$) {
    ServiceLevelDHLGermanyEnum$.inboundSchema = z.nativeEnum(exports.ServiceLevelDHLGermanyEnum);
    ServiceLevelDHLGermanyEnum$.outboundSchema = ServiceLevelDHLGermanyEnum$.inboundSchema;
})(ServiceLevelDHLGermanyEnum$ || (exports.ServiceLevelDHLGermanyEnum$ = ServiceLevelDHLGermanyEnum$ = {}));
/** @internal */
var ServiceLevelDHLeCommerceEnum$;
(function (ServiceLevelDHLeCommerceEnum$) {
    ServiceLevelDHLeCommerceEnum$.inboundSchema = z.nativeEnum(exports.ServiceLevelDHLeCommerceEnum);
    ServiceLevelDHLeCommerceEnum$.outboundSchema = ServiceLevelDHLeCommerceEnum$.inboundSchema;
})(ServiceLevelDHLeCommerceEnum$ || (exports.ServiceLevelDHLeCommerceEnum$ = ServiceLevelDHLeCommerceEnum$ = {}));
/** @internal */
var ServiceLevelDHLExpressEnum$;
(function (ServiceLevelDHLExpressEnum$) {
    ServiceLevelDHLExpressEnum$.inboundSchema = z.nativeEnum(exports.ServiceLevelDHLExpressEnum);
    ServiceLevelDHLExpressEnum$.outboundSchema = ServiceLevelDHLExpressEnum$.inboundSchema;
})(ServiceLevelDHLExpressEnum$ || (exports.ServiceLevelDHLExpressEnum$ = ServiceLevelDHLExpressEnum$ = {}));
/** @internal */
var ServiceLevelPurolatorEnum$;
(function (ServiceLevelPurolatorEnum$) {
    ServiceLevelPurolatorEnum$.inboundSchema = z.nativeEnum(exports.ServiceLevelPurolatorEnum);
    ServiceLevelPurolatorEnum$.outboundSchema = ServiceLevelPurolatorEnum$.inboundSchema;
})(ServiceLevelPurolatorEnum$ || (exports.ServiceLevelPurolatorEnum$ = ServiceLevelPurolatorEnum$ = {}));
/** @internal */
var ServiceLevelColissimoEnum$;
(function (ServiceLevelColissimoEnum$) {
    ServiceLevelColissimoEnum$.inboundSchema = z.nativeEnum(exports.ServiceLevelColissimoEnum);
    ServiceLevelColissimoEnum$.outboundSchema = ServiceLevelColissimoEnum$.inboundSchema;
})(ServiceLevelColissimoEnum$ || (exports.ServiceLevelColissimoEnum$ = ServiceLevelColissimoEnum$ = {}));
/** @internal */
var ServiceLevelCorreosEspanaEnum$;
(function (ServiceLevelCorreosEspanaEnum$) {
    ServiceLevelCorreosEspanaEnum$.inboundSchema = z.nativeEnum(exports.ServiceLevelCorreosEspanaEnum);
    ServiceLevelCorreosEspanaEnum$.outboundSchema = ServiceLevelCorreosEspanaEnum$.inboundSchema;
})(ServiceLevelCorreosEspanaEnum$ || (exports.ServiceLevelCorreosEspanaEnum$ = ServiceLevelCorreosEspanaEnum$ = {}));
/** @internal */
var ServiceLevelCouriersPleaseEnum$;
(function (ServiceLevelCouriersPleaseEnum$) {
    ServiceLevelCouriersPleaseEnum$.inboundSchema = z.nativeEnum(exports.ServiceLevelCouriersPleaseEnum);
    ServiceLevelCouriersPleaseEnum$.outboundSchema = ServiceLevelCouriersPleaseEnum$.inboundSchema;
})(ServiceLevelCouriersPleaseEnum$ || (exports.ServiceLevelCouriersPleaseEnum$ = ServiceLevelCouriersPleaseEnum$ = {}));
/** @internal */
var ServiceLevelChronopostEnum$;
(function (ServiceLevelChronopostEnum$) {
    ServiceLevelChronopostEnum$.inboundSchema = z.nativeEnum(exports.ServiceLevelChronopostEnum);
    ServiceLevelChronopostEnum$.outboundSchema = ServiceLevelChronopostEnum$.inboundSchema;
})(ServiceLevelChronopostEnum$ || (exports.ServiceLevelChronopostEnum$ = ServiceLevelChronopostEnum$ = {}));
/** @internal */
var ServiceLevelCDLEnum$;
(function (ServiceLevelCDLEnum$) {
    ServiceLevelCDLEnum$.inboundSchema = z.nativeEnum(exports.ServiceLevelCDLEnum);
    ServiceLevelCDLEnum$.outboundSchema = ServiceLevelCDLEnum$.inboundSchema;
})(ServiceLevelCDLEnum$ || (exports.ServiceLevelCDLEnum$ = ServiceLevelCDLEnum$ = {}));
/** @internal */
var ServiceLevelCanadaPostEnum$;
(function (ServiceLevelCanadaPostEnum$) {
    ServiceLevelCanadaPostEnum$.inboundSchema = z.nativeEnum(exports.ServiceLevelCanadaPostEnum);
    ServiceLevelCanadaPostEnum$.outboundSchema = ServiceLevelCanadaPostEnum$.inboundSchema;
})(ServiceLevelCanadaPostEnum$ || (exports.ServiceLevelCanadaPostEnum$ = ServiceLevelCanadaPostEnum$ = {}));
/** @internal */
var ServiceLevelBetterTrucksEnum$;
(function (ServiceLevelBetterTrucksEnum$) {
    ServiceLevelBetterTrucksEnum$.inboundSchema = z.nativeEnum(exports.ServiceLevelBetterTrucksEnum);
    ServiceLevelBetterTrucksEnum$.outboundSchema = ServiceLevelBetterTrucksEnum$.inboundSchema;
})(ServiceLevelBetterTrucksEnum$ || (exports.ServiceLevelBetterTrucksEnum$ = ServiceLevelBetterTrucksEnum$ = {}));
/** @internal */
var ServiceLevelAPGEnum$;
(function (ServiceLevelAPGEnum$) {
    ServiceLevelAPGEnum$.inboundSchema = z.nativeEnum(exports.ServiceLevelAPGEnum);
    ServiceLevelAPGEnum$.outboundSchema = ServiceLevelAPGEnum$.inboundSchema;
})(ServiceLevelAPGEnum$ || (exports.ServiceLevelAPGEnum$ = ServiceLevelAPGEnum$ = {}));
/** @internal */
var ServiceLevelAustraliaPostEnum$;
(function (ServiceLevelAustraliaPostEnum$) {
    ServiceLevelAustraliaPostEnum$.inboundSchema = z.nativeEnum(exports.ServiceLevelAustraliaPostEnum);
    ServiceLevelAustraliaPostEnum$.outboundSchema = ServiceLevelAustraliaPostEnum$.inboundSchema;
})(ServiceLevelAustraliaPostEnum$ || (exports.ServiceLevelAustraliaPostEnum$ = ServiceLevelAustraliaPostEnum$ = {}));
/** @internal */
var ServiceLevelAsendiaEnum$;
(function (ServiceLevelAsendiaEnum$) {
    ServiceLevelAsendiaEnum$.inboundSchema = z.nativeEnum(exports.ServiceLevelAsendiaEnum);
    ServiceLevelAsendiaEnum$.outboundSchema = ServiceLevelAsendiaEnum$.inboundSchema;
})(ServiceLevelAsendiaEnum$ || (exports.ServiceLevelAsendiaEnum$ = ServiceLevelAsendiaEnum$ = {}));
/** @internal */
var ServiceLevelAPCPostalEnum$;
(function (ServiceLevelAPCPostalEnum$) {
    ServiceLevelAPCPostalEnum$.inboundSchema = z.nativeEnum(exports.ServiceLevelAPCPostalEnum);
    ServiceLevelAPCPostalEnum$.outboundSchema = ServiceLevelAPCPostalEnum$.inboundSchema;
})(ServiceLevelAPCPostalEnum$ || (exports.ServiceLevelAPCPostalEnum$ = ServiceLevelAPCPostalEnum$ = {}));
/** @internal */
var ServiceLevelAirterraEnum$;
(function (ServiceLevelAirterraEnum$) {
    ServiceLevelAirterraEnum$.inboundSchema = z.nativeEnum(exports.ServiceLevelAirterraEnum);
    ServiceLevelAirterraEnum$.outboundSchema = ServiceLevelAirterraEnum$.inboundSchema;
})(ServiceLevelAirterraEnum$ || (exports.ServiceLevelAirterraEnum$ = ServiceLevelAirterraEnum$ = {}));
/** @internal */
var ServiceLevelUPSEnum$;
(function (ServiceLevelUPSEnum$) {
    ServiceLevelUPSEnum$.inboundSchema = z.nativeEnum(exports.ServiceLevelUPSEnum);
    ServiceLevelUPSEnum$.outboundSchema = ServiceLevelUPSEnum$.inboundSchema;
})(ServiceLevelUPSEnum$ || (exports.ServiceLevelUPSEnum$ = ServiceLevelUPSEnum$ = {}));
/** @internal */
var ServiceLevelFedExEnum$;
(function (ServiceLevelFedExEnum$) {
    ServiceLevelFedExEnum$.inboundSchema = z.nativeEnum(exports.ServiceLevelFedExEnum);
    ServiceLevelFedExEnum$.outboundSchema = ServiceLevelFedExEnum$.inboundSchema;
})(ServiceLevelFedExEnum$ || (exports.ServiceLevelFedExEnum$ = ServiceLevelFedExEnum$ = {}));
/** @internal */
var ServiceLevelUSPSEnum$;
(function (ServiceLevelUSPSEnum$) {
    ServiceLevelUSPSEnum$.inboundSchema = z.nativeEnum(exports.ServiceLevelUSPSEnum);
    ServiceLevelUSPSEnum$.outboundSchema = ServiceLevelUSPSEnum$.inboundSchema;
})(ServiceLevelUSPSEnum$ || (exports.ServiceLevelUSPSEnum$ = ServiceLevelUSPSEnum$ = {}));
/** @internal */
var ServiceLevelEnumSet$;
(function (ServiceLevelEnumSet$) {
    ServiceLevelEnumSet$.inboundSchema = z.union([
        ServiceLevelUSPSEnum$.inboundSchema,
        ServiceLevelFedExEnum$.inboundSchema,
        ServiceLevelUPSEnum$.inboundSchema,
        ServiceLevelAirterraEnum$.inboundSchema,
        ServiceLevelAPCPostalEnum$.inboundSchema,
        ServiceLevelAsendiaEnum$.inboundSchema,
        ServiceLevelAustraliaPostEnum$.inboundSchema,
        ServiceLevelAPGEnum$.inboundSchema,
        ServiceLevelBetterTrucksEnum$.inboundSchema,
        ServiceLevelCanadaPostEnum$.inboundSchema,
        ServiceLevelCDLEnum$.inboundSchema,
        ServiceLevelChronopostEnum$.inboundSchema,
        ServiceLevelCouriersPleaseEnum$.inboundSchema,
        ServiceLevelCorreosEspanaEnum$.inboundSchema,
        ServiceLevelColissimoEnum$.inboundSchema,
        ServiceLevelPurolatorEnum$.inboundSchema,
        ServiceLevelDHLExpressEnum$.inboundSchema,
        ServiceLevelDHLeCommerceEnum$.inboundSchema,
        ServiceLevelDHLGermanyEnum$.inboundSchema,
        ServiceLevelDPDDEEnum$.inboundSchema,
        ServiceLevelDPDUKEnum$.inboundSchema,
        ServiceLevelDeutschePostEnum$.inboundSchema,
        ServiceLevelAramexAustraliaEnum$.inboundSchema,
        ServiceLevelGlobegisticsEnum$.inboundSchema,
        ServiceLevelGLSUSEnum$.inboundSchema,
        ServiceLevelLSOEnum$.inboundSchema,
        ServiceLevelMondialRelayEnum$.inboundSchema,
        ServiceLevelMaergoEnum$.inboundSchema,
        ServiceLevelParcelforceEnum$.inboundSchema,
        ServiceLevelPostItalianeEnum$.inboundSchema,
        ServiceLevelePostGlobalEnum$.inboundSchema,
        ServiceLevelRoyalMailEnum$.inboundSchema,
        ServiceLevelSendleEnum$.inboundSchema,
        ServiceLevelOnTracEnum$.inboundSchema,
        ServiceLevelJitsuEnum$.inboundSchema,
        ServiceLevelLasershipEnum$.inboundSchema,
        ServiceLevelEvriUKEnum$.inboundSchema,
        ServiceLevelUDSEnum$.inboundSchema,
        ServiceLevelVehoEnum$.inboundSchema,
        ServiceLevelSwyftEnum$.inboundSchema,
    ]);
    ServiceLevelEnumSet$.outboundSchema = z.union([
        ServiceLevelUSPSEnum$.outboundSchema,
        ServiceLevelFedExEnum$.outboundSchema,
        ServiceLevelUPSEnum$.outboundSchema,
        ServiceLevelAirterraEnum$.outboundSchema,
        ServiceLevelAPCPostalEnum$.outboundSchema,
        ServiceLevelAsendiaEnum$.outboundSchema,
        ServiceLevelAustraliaPostEnum$.outboundSchema,
        ServiceLevelAPGEnum$.outboundSchema,
        ServiceLevelBetterTrucksEnum$.outboundSchema,
        ServiceLevelCanadaPostEnum$.outboundSchema,
        ServiceLevelCDLEnum$.outboundSchema,
        ServiceLevelChronopostEnum$.outboundSchema,
        ServiceLevelCouriersPleaseEnum$.outboundSchema,
        ServiceLevelCorreosEspanaEnum$.outboundSchema,
        ServiceLevelColissimoEnum$.outboundSchema,
        ServiceLevelPurolatorEnum$.outboundSchema,
        ServiceLevelDHLExpressEnum$.outboundSchema,
        ServiceLevelDHLeCommerceEnum$.outboundSchema,
        ServiceLevelDHLGermanyEnum$.outboundSchema,
        ServiceLevelDPDDEEnum$.outboundSchema,
        ServiceLevelDPDUKEnum$.outboundSchema,
        ServiceLevelDeutschePostEnum$.outboundSchema,
        ServiceLevelAramexAustraliaEnum$.outboundSchema,
        ServiceLevelGlobegisticsEnum$.outboundSchema,
        ServiceLevelGLSUSEnum$.outboundSchema,
        ServiceLevelLSOEnum$.outboundSchema,
        ServiceLevelMondialRelayEnum$.outboundSchema,
        ServiceLevelMaergoEnum$.outboundSchema,
        ServiceLevelParcelforceEnum$.outboundSchema,
        ServiceLevelPostItalianeEnum$.outboundSchema,
        ServiceLevelePostGlobalEnum$.outboundSchema,
        ServiceLevelRoyalMailEnum$.outboundSchema,
        ServiceLevelSendleEnum$.outboundSchema,
        ServiceLevelOnTracEnum$.outboundSchema,
        ServiceLevelJitsuEnum$.outboundSchema,
        ServiceLevelLasershipEnum$.outboundSchema,
        ServiceLevelEvriUKEnum$.outboundSchema,
        ServiceLevelUDSEnum$.outboundSchema,
        ServiceLevelVehoEnum$.outboundSchema,
        ServiceLevelSwyftEnum$.outboundSchema,
    ]);
})(ServiceLevelEnumSet$ || (exports.ServiceLevelEnumSet$ = ServiceLevelEnumSet$ = {}));
