// shippo/models/components/servicelevelwithparent.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceLevelWithParent$ = exports.ParentServicelevel$ = void 0;
const primitives_js_1 = require("../../lib/primitives.js");
const z = __importStar(require("zod"));
/** @internal */
var ParentServicelevel$;
(function (ParentServicelevel$) {
    ParentServicelevel$.inboundSchema = z
        .object({
        name: z.string().optional(),
        terms: z.string().optional(),
        token: z.string().optional(),
        extended_token: z.string().optional(),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            extended_token: "extendedToken",
        });
    });
    ParentServicelevel$.outboundSchema = z
        .object({
        name: z.string().optional(),
        terms: z.string().optional(),
        token: z.string().optional(),
        extendedToken: z.string().optional(),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            extendedToken: "extended_token",
        });
    });
})(ParentServicelevel$ || (exports.ParentServicelevel$ = ParentServicelevel$ = {}));
/** @internal */
var ServiceLevelWithParent$;
(function (ServiceLevelWithParent$) {
    ServiceLevelWithParent$.inboundSchema = z
        .object({
        name: z.string().optional(),
        terms: z.string().optional(),
        token: z.string().optional(),
        extended_token: z.string().optional(),
        parent_servicelevel: z.lazy(() => ParentServicelevel$.inboundSchema).optional(),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            extended_token: "extendedToken",
            parent_servicelevel: "parentServicelevel",
        });
    });
    ServiceLevelWithParent$.outboundSchema = z
        .object({
        name: z.string().optional(),
        terms: z.string().optional(),
        token: z.string().optional(),
        extendedToken: z.string().optional(),
        parentServicelevel: z.lazy(() => ParentServicelevel$.outboundSchema).optional(),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            extendedToken: "extended_token",
            parentServicelevel: "parent_servicelevel",
        });
    });
})(ServiceLevelWithParent$ || (exports.ServiceLevelWithParent$ = ServiceLevelWithParent$ = {}));
