// shippo/models/components/shipmentcreaterequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ShipmentCreateRequest$ = exports.Parcels$ = exports.ShipmentCreateRequestCustomsDeclaration$ = exports.AddressTo$ = exports.AddressReturn$ = exports.AddressFrom$ = void 0;
const primitives_js_1 = require("../../lib/primitives.js");
const addresscreaterequest_js_1 = require("./addresscreaterequest.js");
const customsdeclarationcreaterequest_js_1 = require("./customsdeclarationcreaterequest.js");
const parcelcreatefromtemplaterequest_js_1 = require("./parcelcreatefromtemplaterequest.js");
const parcelcreaterequest_js_1 = require("./parcelcreaterequest.js");
const shipmentextra_js_1 = require("./shipmentextra.js");
const z = __importStar(require("zod"));
/** @internal */
var AddressFrom$;
(function (AddressFrom$) {
    AddressFrom$.inboundSchema = z.union([
        addresscreaterequest_js_1.AddressCreateRequest$.inboundSchema,
        z.string(),
    ]);
    AddressFrom$.outboundSchema = z.union([
        addresscreaterequest_js_1.AddressCreateRequest$.outboundSchema,
        z.string(),
    ]);
})(AddressFrom$ || (exports.AddressFrom$ = AddressFrom$ = {}));
/** @internal */
var AddressReturn$;
(function (AddressReturn$) {
    AddressReturn$.inboundSchema = z.union([
        addresscreaterequest_js_1.AddressCreateRequest$.inboundSchema,
        z.string(),
    ]);
    AddressReturn$.outboundSchema = z.union([
        addresscreaterequest_js_1.AddressCreateRequest$.outboundSchema,
        z.string(),
    ]);
})(AddressReturn$ || (exports.AddressReturn$ = AddressReturn$ = {}));
/** @internal */
var AddressTo$;
(function (AddressTo$) {
    AddressTo$.inboundSchema = z.union([
        addresscreaterequest_js_1.AddressCreateRequest$.inboundSchema,
        z.string(),
    ]);
    AddressTo$.outboundSchema = z.union([
        addresscreaterequest_js_1.AddressCreateRequest$.outboundSchema,
        z.string(),
    ]);
})(AddressTo$ || (exports.AddressTo$ = AddressTo$ = {}));
/** @internal */
var ShipmentCreateRequestCustomsDeclaration$;
(function (ShipmentCreateRequestCustomsDeclaration$) {
    ShipmentCreateRequestCustomsDeclaration$.inboundSchema = z.union([customsdeclarationcreaterequest_js_1.CustomsDeclarationCreateRequest$.inboundSchema, z.string()]);
    ShipmentCreateRequestCustomsDeclaration$.outboundSchema = z.union([customsdeclarationcreaterequest_js_1.CustomsDeclarationCreateRequest$.outboundSchema, z.string()]);
})(ShipmentCreateRequestCustomsDeclaration$ || (exports.ShipmentCreateRequestCustomsDeclaration$ = ShipmentCreateRequestCustomsDeclaration$ = {}));
/** @internal */
var Parcels$;
(function (Parcels$) {
    Parcels$.inboundSchema = z.union([
        parcelcreatefromtemplaterequest_js_1.ParcelCreateFromTemplateRequest$.inboundSchema,
        parcelcreaterequest_js_1.ParcelCreateRequest$.inboundSchema,
        z.string(),
    ]);
    Parcels$.outboundSchema = z.union([
        parcelcreatefromtemplaterequest_js_1.ParcelCreateFromTemplateRequest$.outboundSchema,
        parcelcreaterequest_js_1.ParcelCreateRequest$.outboundSchema,
        z.string(),
    ]);
})(Parcels$ || (exports.Parcels$ = Parcels$ = {}));
/** @internal */
var ShipmentCreateRequest$;
(function (ShipmentCreateRequest$) {
    ShipmentCreateRequest$.inboundSchema = z
        .object({
        extra: shipmentextra_js_1.ShipmentExtra$.inboundSchema.optional(),
        metadata: z.string().optional(),
        shipment_date: z.string().optional(),
        address_from: z.union([addresscreaterequest_js_1.AddressCreateRequest$.inboundSchema, z.string()]),
        address_return: z.union([addresscreaterequest_js_1.AddressCreateRequest$.inboundSchema, z.string()]).optional(),
        address_to: z.union([addresscreaterequest_js_1.AddressCreateRequest$.inboundSchema, z.string()]),
        customs_declaration: z
            .union([customsdeclarationcreaterequest_js_1.CustomsDeclarationCreateRequest$.inboundSchema, z.string()])
            .optional(),
        async: z.boolean().optional(),
        carrier_accounts: z.array(z.string()).optional(),
        parcels: z.array(z.union([
            parcelcreatefromtemplaterequest_js_1.ParcelCreateFromTemplateRequest$.inboundSchema,
            parcelcreaterequest_js_1.ParcelCreateRequest$.inboundSchema,
            z.string(),
        ])),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            shipment_date: "shipmentDate",
            address_from: "addressFrom",
            address_return: "addressReturn",
            address_to: "addressTo",
            customs_declaration: "customsDeclaration",
            carrier_accounts: "carrierAccounts",
        });
    });
    ShipmentCreateRequest$.outboundSchema = z
        .object({
        extra: shipmentextra_js_1.ShipmentExtra$.outboundSchema.optional(),
        metadata: z.string().optional(),
        shipmentDate: z.string().optional(),
        addressFrom: z.union([addresscreaterequest_js_1.AddressCreateRequest$.outboundSchema, z.string()]),
        addressReturn: z.union([addresscreaterequest_js_1.AddressCreateRequest$.outboundSchema, z.string()]).optional(),
        addressTo: z.union([addresscreaterequest_js_1.AddressCreateRequest$.outboundSchema, z.string()]),
        customsDeclaration: z
            .union([customsdeclarationcreaterequest_js_1.CustomsDeclarationCreateRequest$.outboundSchema, z.string()])
            .optional(),
        async: z.boolean().optional(),
        carrierAccounts: z.array(z.string()).optional(),
        parcels: z.array(z.union([
            parcelcreatefromtemplaterequest_js_1.ParcelCreateFromTemplateRequest$.outboundSchema,
            parcelcreaterequest_js_1.ParcelCreateRequest$.outboundSchema,
            z.string(),
        ])),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            shipmentDate: "shipment_date",
            addressFrom: "address_from",
            addressReturn: "address_return",
            addressTo: "address_to",
            customsDeclaration: "customs_declaration",
            carrierAccounts: "carrier_accounts",
        });
    });
})(ShipmentCreateRequest$ || (exports.ShipmentCreateRequest$ = ShipmentCreateRequest$ = {}));
