// shippo/models/components/transactioncreaterequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransactionCreateRequest$ = void 0;
const primitives_js_1 = require("../../lib/primitives.js");
const labelfiletypeenum_js_1 = require("./labelfiletypeenum.js");
const z = __importStar(require("zod"));
/** @internal */
var TransactionCreateRequest$;
(function (TransactionCreateRequest$) {
    TransactionCreateRequest$.inboundSchema = z
        .object({
        async: z.boolean().default(true),
        label_file_type: labelfiletypeenum_js_1.LabelFileTypeEnum$.inboundSchema.optional(),
        metadata: z.string().optional(),
        rate: z.string(),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            label_file_type: "labelFileType",
        });
    });
    TransactionCreateRequest$.outboundSchema = z
        .object({
        async: z.boolean().default(true),
        labelFileType: labelfiletypeenum_js_1.LabelFileTypeEnum$.outboundSchema.optional(),
        metadata: z.string().optional(),
        rate: z.string(),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            labelFileType: "label_file_type",
        });
    });
})(TransactionCreateRequest$ || (exports.TransactionCreateRequest$ = TransactionCreateRequest$ = {}));
