// shippo/models/components/transaction.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Transaction$ = exports.TransactionRate$ = exports.CreatedBy$ = void 0;
const primitives_js_1 = require("../../lib/primitives.js");
const corerate_js_1 = require("./corerate.js");
const labelfiletypeenum_js_1 = require("./labelfiletypeenum.js");
const objectstateenum_js_1 = require("./objectstateenum.js");
const responsemessage_js_1 = require("./responsemessage.js");
const trackingstatusenum_js_1 = require("./trackingstatusenum.js");
const transactionstatusenum_js_1 = require("./transactionstatusenum.js");
const z = __importStar(require("zod"));
/** @internal */
var CreatedBy$;
(function (CreatedBy$) {
    CreatedBy$.inboundSchema = z
        .object({
        first_name: z.string().optional(),
        last_name: z.string().optional(),
        username: z.string().optional(),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            first_name: "firstName",
            last_name: "lastName",
        });
    });
    CreatedBy$.outboundSchema = z
        .object({
        firstName: z.string().optional(),
        lastName: z.string().optional(),
        username: z.string().optional(),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            firstName: "first_name",
            lastName: "last_name",
        });
    });
})(CreatedBy$ || (exports.CreatedBy$ = CreatedBy$ = {}));
/** @internal */
var TransactionRate$;
(function (TransactionRate$) {
    TransactionRate$.inboundSchema = z.union([
        corerate_js_1.CoreRate$.inboundSchema,
        z.string(),
    ]);
    TransactionRate$.outboundSchema = z.union([
        corerate_js_1.CoreRate$.outboundSchema,
        z.string(),
    ]);
})(TransactionRate$ || (exports.TransactionRate$ = TransactionRate$ = {}));
/** @internal */
var Transaction$;
(function (Transaction$) {
    Transaction$.inboundSchema = z
        .object({
        commercial_invoice_url: z.string().optional(),
        created_by: z.nullable(z.lazy(() => CreatedBy$.inboundSchema)).optional(),
        eta: z.string().optional(),
        label_file_type: labelfiletypeenum_js_1.LabelFileTypeEnum$.inboundSchema.optional(),
        label_url: z.string().optional(),
        messages: z.array(responsemessage_js_1.ResponseMessage$.inboundSchema).optional(),
        metadata: z.string().optional(),
        object_created: z
            .string()
            .datetime({ offset: true })
            .transform((v) => new Date(v))
            .optional(),
        object_id: z.string().optional(),
        object_owner: z.string().optional(),
        object_state: objectstateenum_js_1.ObjectStateEnum$.inboundSchema.optional(),
        object_updated: z
            .string()
            .datetime({ offset: true })
            .transform((v) => new Date(v))
            .optional(),
        parcel: z.string().optional(),
        qr_code_url: z.string().optional(),
        rate: z.union([corerate_js_1.CoreRate$.inboundSchema, z.string()]).optional(),
        status: transactionstatusenum_js_1.TransactionStatusEnum$.inboundSchema.optional(),
        test: z.boolean().optional(),
        tracking_number: z.string().optional(),
        tracking_status: trackingstatusenum_js_1.TrackingStatusEnum$.inboundSchema.optional(),
        tracking_url_provider: z.string().optional(),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            commercial_invoice_url: "commercialInvoiceUrl",
            created_by: "createdBy",
            label_file_type: "labelFileType",
            label_url: "labelUrl",
            object_created: "objectCreated",
            object_id: "objectId",
            object_owner: "objectOwner",
            object_state: "objectState",
            object_updated: "objectUpdated",
            qr_code_url: "qrCodeUrl",
            tracking_number: "trackingNumber",
            tracking_status: "trackingStatus",
            tracking_url_provider: "trackingUrlProvider",
        });
    });
    Transaction$.outboundSchema = z
        .object({
        commercialInvoiceUrl: z.string().optional(),
        createdBy: z.nullable(z.lazy(() => CreatedBy$.outboundSchema)).optional(),
        eta: z.string().optional(),
        labelFileType: labelfiletypeenum_js_1.LabelFileTypeEnum$.outboundSchema.optional(),
        labelUrl: z.string().optional(),
        messages: z.array(responsemessage_js_1.ResponseMessage$.outboundSchema).optional(),
        metadata: z.string().optional(),
        objectCreated: z
            .date()
            .transform((v) => v.toISOString())
            .optional(),
        objectId: z.string().optional(),
        objectOwner: z.string().optional(),
        objectState: objectstateenum_js_1.ObjectStateEnum$.outboundSchema.optional(),
        objectUpdated: z
            .date()
            .transform((v) => v.toISOString())
            .optional(),
        parcel: z.string().optional(),
        qrCodeUrl: z.string().optional(),
        rate: z.union([corerate_js_1.CoreRate$.outboundSchema, z.string()]).optional(),
        status: transactionstatusenum_js_1.TransactionStatusEnum$.outboundSchema.optional(),
        test: z.boolean().optional(),
        trackingNumber: z.string().optional(),
        trackingStatus: trackingstatusenum_js_1.TrackingStatusEnum$.outboundSchema.optional(),
        trackingUrlProvider: z.string().optional(),
    })
        .transform((v) => {
        return (0, primitives_js_1.remap)(v, {
            commercialInvoiceUrl: "commercial_invoice_url",
            createdBy: "created_by",
            labelFileType: "label_file_type",
            labelUrl: "label_url",
            objectCreated: "object_created",
            objectId: "object_id",
            objectOwner: "object_owner",
            objectState: "object_state",
            objectUpdated: "object_updated",
            qrCodeUrl: "qr_code_url",
            trackingNumber: "tracking_number",
            trackingStatus: "tracking_status",
            trackingUrlProvider: "tracking_url_provider",
        });
    });
})(Transaction$ || (exports.Transaction$ = Transaction$ = {}));
